<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Batch correction of phosphoproteomics dataset with PhosR • PhosR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Batch correction of phosphoproteomics dataset with PhosR">
<meta property="og:description" content="PhosR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">PhosR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    PhosR
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/PhosR.html">Introduction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Preprocessing
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/web/imputation.html">Imputation</a>
    </li>
    <li>
      <a href="../../articles/web/batch_correction.html">Batch correction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Downstream analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/web/pathway_analysis.html">Pathway analysis</a>
    </li>
    <li>
      <a href="../../articles/web/site_gene_analysis.html">Site- and gene-centric analysis</a>
    </li>
    <li>
      <a href="../../articles/web/signalomes.html">Signalomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PYangLab/PhosR">
    <span class="fas fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="batch_correction_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Batch correction of phosphoproteomics dataset with PhosR</h1>
            
            <h4 class="date">12 February 2021</h4>
      
      
      <div class="hidden name"><code>batch_correction.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>A common but largely unaddressed challenge in phosphoproteomic data analysis is to correct for batch effect. Without correcting for batch effect, it is often not possible to analyze datasets in an integrative manner. To perform data integration and batch effect correction, we identified a set of stably phosphorylated sites (SPSs) across a panel of phosphoproteomic datasets and, using these SPSs, implemented a wrapper function of RUV-III from the <code>ruv</code> package called <code>RUVphospho</code>.</p>
<p>Note that when the input data contains missing values, imputation should be performed before batch correction since RUV-III requires a complete data matrix. The imputed values are removed by default after normalisation but can be retained for downstream analysis if the users wish to use the imputed matrix. This vignette will provide an example of how PhosR can be used for batch correction.</p>
</div>
<div id="loading-packages-and-data" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-packages-and-data" class="anchor"></a>Loading packages and data</h1>
<p>If you haven’t already done so, load the PhosR package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PhosR</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://stringr.tidyverse.org">stringr</a></span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p>In this example, we will use L6 myotube phosphoproteome dataset (with accession number PXD019127) and the SPSs we identified from a panel of phosphoproteomic datasets (please refer to <a href="https://www.biorxiv.org/content/10.1101/2020.08.31.276329v2.full">our preprint</a> for the full list of the datasets used). The <code>SPSs</code> will be used as our <code>negative control</code> in RUV normalisation.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho_L6_ratio_pe"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"SPSs"</span><span class="op">)</span>

<span class="va">ppe</span> <span class="op">&lt;-</span> <span class="va">phospho.L6.ratio.pe</span>
<span class="va">ppe</span>
<span class="co">#&gt; class: PhosphoExperiment </span>
<span class="co">#&gt; dim: 6654 12 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): Quantification</span>
<span class="co">#&gt; rownames(6654): D3ZNS8;AAAS;S495;THIPLYFVNAQFPRFSPVLGRAQEPPAGGGG</span>
<span class="co">#&gt;   Q9R0Z7;AAGAB;S210;RSVGSAESCQCEQEPSPTAERTESLPGHRSG ...</span>
<span class="co">#&gt;   D3ZG78;ZZEF1;S1516;SGPSAAEVSTAEEPSSPSTPTRRPPFTRGRL</span>
<span class="co">#&gt;   D3ZG78;ZZEF1;S1535;PTRRPPFTRGRLRLLSFRSMEETRPVPTVKE</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(12): AICAR_exp1 AICAR_exp2 ... AICARIns_exp3 AICARIns_exp4</span>
<span class="co">#&gt; colData names(0):</span></code></pre></div>
</div>
<div id="setting-up-the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#setting-up-the-data" class="anchor"></a>Setting up the data</h1>
<p>The L6 myotube data contains phosphoproteomic samples from three treatment conditions each with quadruplicates. Myotube cells were treated with either AICAR or Insulin (Ins), which are both important modulators of the insulin signalling pathway, or both (AICARIns) before phosphoproteomic analysis.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"AICAR_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; [1] "AICAR_exp1" "AICAR_exp2" "AICAR_exp3" "AICAR_exp4"</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"^Ins_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; [1] "Ins_exp1" "Ins_exp2" "Ins_exp3" "Ins_exp4"</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"AICARIns_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; [1] "AICARIns_exp1" "AICARIns_exp2" "AICARIns_exp3" "AICARIns_exp4"</span></code></pre></div>
<p>Note that we have in total 6654 quantified phosphosites and 12 samples in total.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>
<span class="co">#&gt; [1] 6654   12</span></code></pre></div>
<p>We have already performed the relevant processing steps to generate a dense matrix. Please refer to the <code>imputation</code> page to perform filtering and imputation of phosphosites in order to generate a matrix without any missing values.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 0</span></code></pre></div>
<p>We will extract phosphosite labels.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sites</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">GeneSymbol</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,<span class="st">";"</span>,
                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Residue</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,
                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Site</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,
                 <span class="st">";"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span></code></pre></div>
<p>Lastly, we will take the grouping information from <code>colnames</code> of our matrix.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># take the grouping information</span>
<span class="va">grps</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span>
<span class="va">grps</span>
<span class="co">#&gt;  [1] "AICAR"    "AICAR"    "AICAR"    "AICAR"    "Ins"      "Ins"     </span>
<span class="co">#&gt;  [7] "Ins"      "Ins"      "AICARIns" "AICARIns" "AICARIns" "AICARIns"</span></code></pre></div>
</div>
<div id="diagnosing-batch-effect" class="section level1">
<h1 class="hasAnchor">
<a href="#diagnosing-batch-effect" class="anchor"></a>Diagnosing batch effect</h1>
<p>There are a number of ways to diagnose batch effect. In <code>PhosR</code>, we make use of two visualisation methods to detect batch effect: dendrogram of hierarchical clustering and a principal component analysis (PCA) plot. We use the <code>plotQC</code> function we introduced in the <em>imputation</em> section of the vignette.</p>
<p>By setting <em>panel = dendrogram</em>, we can plot the dendrogram illustrating the results of unsupervised hierarchical clustering of our 12 samples. Clustering results of the samples demonstrate that there is a strong batch effect (where batch denoted as <em>expX</em>, where X refers to the batch number). This is particularly evident for samples from <em>Ins</em> and <em>AICARIns</em> treated conditions.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, panel <span class="op">=</span> <span class="st">"dendrogram"</span>, grps<span class="op">=</span><span class="va">grps</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Before batch correction"</span><span class="op">)</span></code></pre></div>
<p><img src="batch_correction_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<p>We can also visualise the samples in PCA space by setting <em>panel = pca</em>. The PCA plot demonstrates aggregation of samples by batch rather than treatment groups (each point represents a sample coloured by treatment condition). It has become clearer that even within the <em>AICAR</em> treated samples, there is some degree of batch effect as data points are separated between samples from batches 1 and 2 and those from batches 3 and 4.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, grps<span class="op">=</span><span class="va">grps</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, panel <span class="op">=</span> <span class="st">"pca"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Before batch correction"</span><span class="op">)</span></code></pre></div>
<p><img src="batch_correction_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
<div id="correcting-batch-effect" class="section level1">
<h1 class="hasAnchor">
<a href="#correcting-batch-effect" class="anchor"></a>Correcting batch effect</h1>
<p>We have now diagnosed that our dataset exhibits batch effect that is driven by experiment runs for samples treated with three different conditions. To address this batch effect, we correct for this unwanted variation in the data by utilising our pre-defined SPSs as a negative control for <code>RUVphospho</code>.</p>
<p>First, we construct a design matrix by condition.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">design</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">grps</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">design</span>
<span class="co">#&gt;    grpsAICAR grpsAICARIns grpsIns</span>
<span class="co">#&gt; 1          1            0       0</span>
<span class="co">#&gt; 2          1            0       0</span>
<span class="co">#&gt; 3          1            0       0</span>
<span class="co">#&gt; 4          1            0       0</span>
<span class="co">#&gt; 5          0            0       1</span>
<span class="co">#&gt; 6          0            0       1</span>
<span class="co">#&gt; 7          0            0       1</span>
<span class="co">#&gt; 8          0            0       1</span>
<span class="co">#&gt; 9          0            1       0</span>
<span class="co">#&gt; 10         0            1       0</span>
<span class="co">#&gt; 11         0            1       0</span>
<span class="co">#&gt; 12         0            1       0</span>
<span class="co">#&gt; attr(,"assign")</span>
<span class="co">#&gt; [1] 1 1 1</span>
<span class="co">#&gt; attr(,"contrasts")</span>
<span class="co">#&gt; attr(,"contrasts")$grps</span>
<span class="co">#&gt; [1] "contr.treatment"</span></code></pre></div>
<p>We will then use the <code>RUVphospho</code> function to normalise the data. Besides the quantification matrix and the design matrix, there are two other important inputs to <code>RUVphospho</code>: 1) the <em>ctl</em> argument is an integer vector denoting the position of SPSs within the quantification matrix 2) <em>k</em> parameter is an integer denoting the expected number of experimental (e.g., <em>treatment</em>) groups within the data</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># phosphoproteomics data normalisation and batch correction using RUV</span>
<span class="va">ctl</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">sites</span> <span class="op">%in%</span> <span class="va">SPSs</span><span class="op">)</span>
<span class="va">ppe</span> <span class="op">=</span> <span class="fu"><a href="../../reference/RUVphospho.html">RUVphospho</a></span><span class="op">(</span><span class="va">ppe</span>, M <span class="op">=</span> <span class="va">design</span>, k <span class="op">=</span> <span class="fl">3</span>, ctl <span class="op">=</span> <span class="va">ctl</span><span class="op">)</span></code></pre></div>
</div>
<div id="quality-control" class="section level1">
<h1 class="hasAnchor">
<a href="#quality-control" class="anchor"></a>Quality control</h1>
<p>As quality control, we will demonstrate and evaluate our normalisation method with hierarchical clustering and PCA plot using again <code>plotQC</code>. Both the hierarchical clustering and PCA results demonstrate the normalisation procedure in <code>PhosR</code> facilitates effective batch correction.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot after batch correction</span>
<span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="../../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, grps<span class="op">=</span><span class="va">grps</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, panel <span class="op">=</span> <span class="st">"dendrogram"</span> <span class="op">)</span>
<span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="../../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">normalised</span>, grps<span class="op">=</span><span class="va">grps</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, panel<span class="op">=</span><span class="st">"dendrogram"</span><span class="op">)</span>
<span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="batch_correction_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="../../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, panel <span class="op">=</span> <span class="st">"pca"</span>, grps<span class="op">=</span><span class="va">grps</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Before Batch correction"</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="../../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">normalised</span>, grps<span class="op">=</span><span class="va">grps</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, panel<span class="op">=</span><span class="st">"pca"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"After Batch correction"</span><span class="op">)</span>
<span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, nrow <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="batch_correction_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
</div>
<div id="generating-spss" class="section level1">
<h1 class="hasAnchor">
<a href="#generating-spss" class="anchor"></a>Generating SPSs</h1>
<p>We note that the current <em>SPS</em> are derived from phosphoproteomic datasets derived from mouse cells. To enable users working on phosphoproteomic datasets derived from other species, we have developed a function <code>getSPS</code> that takes in multiple phosphoproteomics datasets stored as phosphoExperiment objects to generate set of <em>SPS</em>. Users can use their in-house or public datasets derived from the same speciies to generate species-specific <em>SPS</em>, which they can use to normalise their data.Below demonstrate the usage of the <code>getSPS</code> function.</p>
<p>First, load datasets to use for SPS generation.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho_L6_ratio_pe"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho_liverInsTC_RUV_pe"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho.cells.Ins.pe"</span><span class="op">)</span>

<span class="va">ppe1</span> <span class="op">&lt;-</span> <span class="va">phospho.L6.ratio.pe</span>
<span class="va">ppe2</span> <span class="op">&lt;-</span> <span class="va">phospho.liver.Ins.TC.ratio.RUV.pe</span>
<span class="va">ppe3</span> <span class="op">&lt;-</span> <span class="va">phospho.cells.Ins.pe</span></code></pre></div>
<p>Filter, impute, and transform ppe3 (other inputs have already been processed).</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">grp3</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">'_[0-9]{1}'</span>, <span class="st">''</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe3</span><span class="op">)</span><span class="op">)</span>
<span class="va">ppe3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/selectGrps.html">selectGrps</a></span><span class="op">(</span><span class="va">ppe3</span>, grps <span class="op">=</span> <span class="va">grp3</span>, <span class="fl">0.5</span>, n<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="va">ppe3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/tImpute.html">tImpute</a></span><span class="op">(</span><span class="va">ppe3</span><span class="op">)</span>
<span class="va">FL83B.ratio</span> <span class="op">&lt;-</span> <span class="va">ppe3</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">imputed</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span><span class="op">(</span><span class="va">ppe3</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"FL83B_Control"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe3</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="va">Hepa.ratio</span> <span class="op">&lt;-</span> <span class="va">ppe3</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">imputed</span><span class="op">[</span>, <span class="fl">13</span><span class="op">:</span><span class="fl">24</span><span class="op">]</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span><span class="op">(</span><span class="va">ppe3</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"Hepa1.6_Control"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe3</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="va">ppe3</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/cbind.html">cbind</a></span><span class="op">(</span><span class="va">FL83B.ratio</span>, <span class="va">Hepa.ratio</span><span class="op">)</span></code></pre></div>
<p>Generate inputs of <code>getSPS</code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ppe.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">ppe1</span>, <span class="va">ppe2</span>, <span class="va">ppe3</span><span class="op">)</span>
<span class="va">cond.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>grp1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe1</span><span class="op">)</span><span class="op">)</span>,
                  grp2 <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_sub.html">str_sub</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe2</span><span class="op">)</span>, end<span class="op">=</span><span class="op">-</span><span class="fl">5</span><span class="op">)</span>,
                  grp3 <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_sub.html">str_sub</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe3</span><span class="op">)</span>, end<span class="op">=</span><span class="op">-</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Finally run <code>getSPS</code> to generate list of SPSs.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">inhouse_SPSs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/getSPS.html">getSPS</a></span><span class="op">(</span><span class="va">ppe.list</span>, conds <span class="op">=</span> <span class="va">cond.list</span>, residueInfo <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">inhouse_SPSs</span><span class="op">)</span>
<span class="co">#&gt; [1] "TMPO;67"    "RBM5;621"   "THOC2;1417" "SRRM1;427"  "SSB;92"    </span>
<span class="co">#&gt; [6] "PPIG;252"</span></code></pre></div>
<p>Run <code>RUVphospho</code> using the newly generateed SPSs. (<strong>NOTE</strong> that we do not expect the results to be reproducible as the exempler SPSs have been generated from dummy datasets that have been heavily filtered.)</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sites</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">GeneSymbol</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,<span class="st">";"</span>,
                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Site</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>

<span class="va">ctl</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">sites</span> <span class="op">%in%</span> <span class="va">inhouse_SPSs</span><span class="op">)</span>
<span class="va">ppe</span> <span class="op">=</span> <span class="fu"><a href="../../reference/RUVphospho.html">RUVphospho</a></span><span class="op">(</span><span class="va">ppe</span>, M <span class="op">=</span> <span class="va">design</span>, k <span class="op">=</span> <span class="fl">3</span>, ctl <span class="op">=</span> <span class="va">ctl</span><span class="op">)</span></code></pre></div>
</div>
<div id="sessioninfo" class="section level1">
<h1 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>SessionInfo</h1>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.3 (2020-10-10)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.7</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] stringr_1.4.0 PhosR_1.1.4  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] bitops_1.0-6                matrixStats_0.58.0         </span>
<span class="co">#&gt;   [3] fs_1.5.0                    RColorBrewer_1.1-2         </span>
<span class="co">#&gt;   [5] rprojroot_2.0.2             GenomeInfoDb_1.26.2        </span>
<span class="co">#&gt;   [7] tools_4.0.3                 backports_1.2.1            </span>
<span class="co">#&gt;   [9] R6_2.5.0                    BiocGenerics_0.36.0        </span>
<span class="co">#&gt;  [11] colorspace_2.0-0            tidyselect_1.1.0           </span>
<span class="co">#&gt;  [13] gridExtra_2.3               GGally_2.1.0               </span>
<span class="co">#&gt;  [15] preprocessCore_1.52.1       curl_4.3                   </span>
<span class="co">#&gt;  [17] compiler_4.0.3              textshaping_0.2.1          </span>
<span class="co">#&gt;  [19] Biobase_2.50.0              network_1.16.1             </span>
<span class="co">#&gt;  [21] desc_1.2.0                  DelayedArray_0.16.1        </span>
<span class="co">#&gt;  [23] ggdendro_0.1.22             labeling_0.4.2             </span>
<span class="co">#&gt;  [25] scales_1.1.1                pkgdown_1.6.1              </span>
<span class="co">#&gt;  [27] systemfonts_1.0.0           digest_0.6.27              </span>
<span class="co">#&gt;  [29] foreign_0.8-80              rmarkdown_2.6              </span>
<span class="co">#&gt;  [31] rio_0.5.16                  XVector_0.30.0             </span>
<span class="co">#&gt;  [33] pkgconfig_2.0.3             htmltools_0.5.1.1          </span>
<span class="co">#&gt;  [35] MatrixGenerics_1.2.1        highr_0.8                  </span>
<span class="co">#&gt;  [37] ruv_0.9.7.1                 limma_3.46.0               </span>
<span class="co">#&gt;  [39] fastmap_1.1.0               readxl_1.3.1               </span>
<span class="co">#&gt;  [41] rlang_0.4.10                GlobalOptions_0.1.2        </span>
<span class="co">#&gt;  [43] farver_2.0.3                shape_1.4.5                </span>
<span class="co">#&gt;  [45] generics_0.1.0              zip_2.1.1                  </span>
<span class="co">#&gt;  [47] dendextend_1.14.0           dplyr_1.0.4                </span>
<span class="co">#&gt;  [49] car_3.0-10                  RCurl_1.98-1.2             </span>
<span class="co">#&gt;  [51] magrittr_2.0.1              GenomeInfoDbData_1.2.4     </span>
<span class="co">#&gt;  [53] Matrix_1.2-18               Rcpp_1.0.6                 </span>
<span class="co">#&gt;  [55] munsell_0.5.0               S4Vectors_0.28.1           </span>
<span class="co">#&gt;  [57] abind_1.4-5                 viridis_0.5.1              </span>
<span class="co">#&gt;  [59] lifecycle_0.2.0             stringi_1.5.3              </span>
<span class="co">#&gt;  [61] yaml_2.2.1                  carData_3.0-4              </span>
<span class="co">#&gt;  [63] MASS_7.3-53                 SummarizedExperiment_1.20.0</span>
<span class="co">#&gt;  [65] zlibbioc_1.36.0             plyr_1.8.6                 </span>
<span class="co">#&gt;  [67] grid_4.0.3                  parallel_4.0.3             </span>
<span class="co">#&gt;  [69] forcats_0.5.1               crayon_1.4.1               </span>
<span class="co">#&gt;  [71] lattice_0.20-41             cowplot_1.1.1              </span>
<span class="co">#&gt;  [73] haven_2.3.1                 circlize_0.4.12            </span>
<span class="co">#&gt;  [75] hms_1.0.0                   knitr_1.31                 </span>
<span class="co">#&gt;  [77] pillar_1.4.7                igraph_1.2.6               </span>
<span class="co">#&gt;  [79] ggpubr_0.4.0                GenomicRanges_1.42.0       </span>
<span class="co">#&gt;  [81] ggsignif_0.6.0              reshape2_1.4.4             </span>
<span class="co">#&gt;  [83] stats4_4.0.3                glue_1.4.2                 </span>
<span class="co">#&gt;  [85] evaluate_0.14               pcaMethods_1.82.0          </span>
<span class="co">#&gt;  [87] data.table_1.13.6           BiocManager_1.30.10        </span>
<span class="co">#&gt;  [89] vctrs_0.3.6                 cellranger_1.1.0           </span>
<span class="co">#&gt;  [91] gtable_0.3.0                purrr_0.3.4                </span>
<span class="co">#&gt;  [93] tidyr_1.1.2                 reshape_0.8.8              </span>
<span class="co">#&gt;  [95] assertthat_0.2.1            cachem_1.0.3               </span>
<span class="co">#&gt;  [97] ggplot2_3.3.3               openxlsx_4.2.3             </span>
<span class="co">#&gt;  [99] xfun_0.21                   broom_0.7.4                </span>
<span class="co">#&gt; [101] e1071_1.7-4                 rstatix_0.6.0              </span>
<span class="co">#&gt; [103] ragg_0.4.1                  class_7.3-17               </span>
<span class="co">#&gt; [105] viridisLite_0.3.0           tibble_3.0.6               </span>
<span class="co">#&gt; [107] pheatmap_1.0.12             memoise_2.0.0              </span>
<span class="co">#&gt; [109] IRanges_2.24.1              ellipsis_0.3.1             </span>
<span class="co">#&gt; [111] BiocStyle_2.18.1</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Pengyi Yang, Taiyun Kim, Hani Jieun Kim.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
