<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Imputation of phosphoproteomics data with PhosR • PhosR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Imputation of phosphoproteomics data with PhosR">
<meta property="og:description" content="PhosR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">PhosR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    PhosR
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/PhosR.html">Introduction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Preprocessing
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/web/imputation.html">Imputation</a>
    </li>
    <li>
      <a href="../../articles/web/batch_correction.html">Batch correction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Downstream analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/web/pathway_analysis.html">Pathway analysis</a>
    </li>
    <li>
      <a href="../../articles/web/site_gene_analysis.html">Site- and gene-centric analysis</a>
    </li>
    <li>
      <a href="../../articles/web/signalomes.html">Signalomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PYangLab/PhosR">
    <span class="fas fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="imputation_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Imputation of phosphoproteomics data with PhosR</h1>
            
            <h4 class="date">14 February 2021</h4>
      
      
      <div class="hidden name"><code>imputation.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><code>PhosR</code> is a package for the all-rounded analysis of phosphoproteomic data from processing to downstream analysis. This vignette will provide a step-by-step workflow of how PhosR can be used to process and analyse a panel of phosphoproteomic datasets. As one of the first steps of data processing, we will begin by performing filtering and imputation of the data with <code>PhosR</code>.</p>
</div>
<div id="loading-packages-and-data" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-packages-and-data" class="anchor"></a>Loading packages and data</h1>
<p>First, we will load the PhosR package. If you already haven’t done so, please install PhosR as instructed in the main page.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PhosR</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<div id="setting-up-the-data" class="section level2">
<h2 class="hasAnchor">
<a href="#setting-up-the-data" class="anchor"></a>Setting up the data</h2>
<p>We assume that you will have the raw data processed using platforms frequently used for mass-spectrometry based proteomics such as MaxQuant. For demonstration purposes, we will take a parts of phosphoproteomic data generated by <a href="doi:10.1038/nbt.3327">Humphrey et al.</a> with accession number PXD001792. The dataset contains the phosphoproteomic quantifications of two mouse liver cell lines (Hepa1.6 and FL38B) that were treated with either PBS (mock) or insulin.</p>
<p>Let us load the PhosphoExperiment (ppe) object.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho.cells.Ins.pe"</span><span class="op">)</span>

<span class="va">ppe</span> <span class="op">&lt;-</span> <span class="va">phospho.cells.Ins.pe</span> 
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>
<span class="co">#&gt; [1] "PhosphoExperiment"</span>
<span class="co">#&gt; attr(,"package")</span>
<span class="co">#&gt; [1] "PhosR"</span></code></pre></div>
<p>A quick glance of the object.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ppe</span>
<span class="co">#&gt; class: PhosphoExperiment </span>
<span class="co">#&gt; dim: 5000 24 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): Quantification</span>
<span class="co">#&gt; rownames(5000): Q7TPV4;MYBBP1A;S1321;PQSALPKKRARLSLVSRSPSLLQSGVKKRRV</span>
<span class="co">#&gt;   Q3UR85;MYRF;S304;PARAPSPPWPPQGPLSPGTGSLPLSIARAQT ...</span>
<span class="co">#&gt;   P28659-4;NA;S18;AFKLDFLPEMMVDHCSLNSSPVSKKMNGTLD</span>
<span class="co">#&gt;   E9Q8I9;FRY;S1380;HNIELVDSRLLLPGSSPSSPEDEVKDREGEV</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(24): Intensity.FL83B_Control_1 Intensity.FL83B_Control_2 ...</span>
<span class="co">#&gt;   Intensity.Hepa1.6_Ins_5 Intensity.Hepa1.6_Ins_6</span>
<span class="co">#&gt; colData names(0):</span></code></pre></div>
<p>We will take the grouping information from <code>colnames</code> of our matrix.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">grps</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_[0-9]{1}"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>For each cell line, there are two conditions (Control vs Insulin-stimulated) and 6 replicates for each condition.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># FL38B</span>
<span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"Intensity."</span>, <span class="st">""</span>, <span class="va">grps</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span>
<span class="co">#&gt;  [1] "FL83B_Control" "FL83B_Control" "FL83B_Control" "FL83B_Control"</span>
<span class="co">#&gt;  [5] "FL83B_Control" "FL83B_Control" "FL83B_Ins"     "FL83B_Ins"    </span>
<span class="co">#&gt;  [9] "FL83B_Ins"     "FL83B_Ins"     "FL83B_Ins"     "FL83B_Ins"</span>
<span class="co"># Hepa1</span>
<span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"Intensity."</span>, <span class="st">""</span>, <span class="va">grps</span><span class="op">)</span><span class="op">[</span><span class="fl">13</span><span class="op">:</span><span class="fl">24</span><span class="op">]</span>
<span class="co">#&gt;  [1] "Hepa1.6_Control" "Hepa1.6_Control" "Hepa1.6_Control" "Hepa1.6_Control"</span>
<span class="co">#&gt;  [5] "Hepa1.6_Control" "Hepa1.6_Control" "Hepa1.6_Ins"     "Hepa1.6_Ins"    </span>
<span class="co">#&gt;  [9] "Hepa1.6_Ins"     "Hepa1.6_Ins"     "Hepa1.6_Ins"     "Hepa1.6_Ins"</span></code></pre></div>
<p>Note that there are in total 24 samples and 5,000 phosphosites profiled.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>
<span class="co">#&gt; [1] 5000   24</span></code></pre></div>
</div>
</div>
<div id="filtering-of-phosphosites" class="section level1">
<h1 class="hasAnchor">
<a href="#filtering-of-phosphosites" class="anchor"></a>Filtering of phosphosites</h1>
<p>Next, we will perform some filtering of phosphosites so that only phosphosites with quantification for at least 50% of the replicates in at least one of the conditions are retained. For this filtering step, we use the <code>selectGrps</code> function. The filtering leaves us with 1,772 phosphosites.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ppe_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/selectGrps.html">selectGrps</a></span><span class="op">(</span><span class="va">ppe</span>, <span class="va">grps</span>, <span class="fl">0.5</span>, n<span class="op">=</span><span class="fl">1</span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">ppe_filtered</span><span class="op">)</span>
<span class="co">#&gt; [1] 1772   24</span></code></pre></div>
<p><code>selectGrps</code> gives you the option to relax the threshold for filtering. The filtering threshold can therefore be optimized for each dataset.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># In cases where you have fewer replicates ( e.g.,triplicates), you may want to select phosphosites quantified in 70% of replicates. </span>
<span class="va">ppe_filtered_v1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/selectGrps.html">selectGrps</a></span><span class="op">(</span><span class="va">ppe</span>, <span class="va">grps</span>, <span class="fl">0.7</span>, n<span class="op">=</span><span class="fl">1</span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">ppe_filtered_v1</span><span class="op">)</span>
<span class="co">#&gt; [1] 1330   24</span></code></pre></div>
</div>
<div id="imputation-of-phosphosites" class="section level1">
<h1 class="hasAnchor">
<a href="#imputation-of-phosphosites" class="anchor"></a>Imputation of phosphosites</h1>
<p>We can proceed to imputation now that we have filtered for suboptimal phosphosites. To take advantage of data structure and experimental design, PhosR provides users with a lot of flexibility for imputation. There are three functions for imputation: <code>scImpute</code>,<code>tInmpute</code>, and <code>ptImpute</code>. Here, we will demonstrate the use of <code>scImpute</code> and <code>ptImpute</code>.</p>
<div id="site--and-condition-specific-imputation" class="section level2">
<h2 class="hasAnchor">
<a href="#site--and-condition-specific-imputation" class="anchor"></a>Site- and condition-specific imputation</h2>
<p>The <code>scImpute</code> function is used for site- and condition-specific imputation. A pre-defined thereshold is used to select phosphosites to impute. Phosphosites with missing values equal to or greater than a predefined value will be imputed by sampling from the empirical normal distribution constructed from the quantification values of phosphosites from the same condition.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">ppe_imputed_tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/scImpute.html">scImpute</a></span><span class="op">(</span><span class="va">ppe_filtered</span>, <span class="fl">0.5</span>, <span class="va">grps</span><span class="op">)</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe_filtered</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<p>In the above example, only phosphosites that are quantified in more than 50% of samples from the same condition will be imputed.</p>
</div>
<div id="paired-tail-based-imputation" class="section level2">
<h2 class="hasAnchor">
<a href="#paired-tail-based-imputation" class="anchor"></a>Paired tail-based imputation</h2>
<p>We then perform paired tail-based imputation on the dataset imputed with <code>scImpute</code>. Paired tail-based imputation performs imputation of phosphosites that have missing values in <em>all</em> replicates in one condition (e.g. in <em>basal</em>) but not in another condition (e.g., in <em>stimulation</em>). This method of imputation ensures that we do not accidentally filter phosphosites that seemingly have low detection rate.</p>
<p>As for <code>scImpute</code>, we can set a predefined threshold to in another condition (e.g. <em>stimulation</em>), the tail-based imputation is applied to impute for the missing values in the first condition.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">ppe_imputed</span> <span class="op">&lt;-</span> <span class="va">ppe_imputed_tmp</span>
<span class="va">ppe_imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/ptImpute.html">ptImpute</a></span><span class="op">(</span><span class="va">ppe_imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">7</span>,<span class="fl">12</span><span class="op">)</span><span class="op">]</span>, 
                                 <span class="va">ppe_imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span>, 
                                 percent1 <span class="op">=</span> <span class="fl">0.6</span>, percent2 <span class="op">=</span> <span class="fl">0</span>, paired <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">ppe_imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">13</span>,<span class="fl">18</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/ptImpute.html">ptImpute</a></span><span class="op">(</span><span class="va">ppe_imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">19</span>,<span class="fl">24</span><span class="op">)</span><span class="op">]</span>, 
                                     <span class="va">ppe_imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">13</span>,<span class="fl">18</span><span class="op">)</span><span class="op">]</span>, 
                                     percent1 <span class="op">=</span> <span class="fl">0.6</span>, percent2 <span class="op">=</span> <span class="fl">0</span>, paired <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>Lastly, we perform normalisation of the filtered and imputed phosphoproteomic data.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ppe_imputed_scaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/medianScaling.html">medianScaling</a></span><span class="op">(</span><span class="va">ppe_imputed</span>, scale <span class="op">=</span> <span class="cn">FALSE</span>, assay <span class="op">=</span> <span class="st">"imputed"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="quantification-plots" class="section level1">
<h1 class="hasAnchor">
<a href="#quantification-plots" class="anchor"></a>Quantification plots</h1>
<p>A useful function in <code>PhosR</code> is to visualize the percentage of quantified sites before and after filtering and imputation. The main inputs of <code>plotQC</code> are the quantification matrix, sample labels (equating the column names of the matrix), an integer indicating the panel to plot, and lastly, a color vector. To visualize the percentage of quantified sites, use the <code>plotQC</code> function and set <em>panel = quantify</em> to visualise bar plots of samples.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="../../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe_filtered</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe_filtered</span><span class="op">)</span>,
        panel <span class="op">=</span> <span class="st">"quantify"</span>, grps <span class="op">=</span> <span class="va">grps</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="../../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe_imputed_scaled</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">scaled</span>, 
       labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe_imputed_scaled</span><span class="op">)</span>, panel <span class="op">=</span> <span class="st">"quantify"</span>, grps <span class="op">=</span> <span class="va">grps</span><span class="op">)</span>
<span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="imputation_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p>By setting <em>panel = dendrogram</em>, we can visualise the results of unsupervised hierarchical clustering of samples as a dendrogram. The dendrogram demonstrates that imputation has improved the clustering of the samples so that replicates from the same conditions cluster together.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="../../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe_filtered</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, 
       labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe_filtered</span><span class="op">)</span>, panel <span class="op">=</span> <span class="st">"dendrogram"</span>, 
       grps <span class="op">=</span> <span class="va">grps</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="../../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe_imputed_scaled</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">scaled</span>, 
       labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe_imputed_scaled</span><span class="op">)</span>, 
       panel <span class="op">=</span> <span class="st">"dendrogram"</span>, grps <span class="op">=</span> <span class="va">grps</span><span class="op">)</span>
<span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="imputation_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>We can now move onto the next step in the <code>PhosR</code> workflow: integration of datasets and batch correction.</p>
</div>
<div id="sessioninfo" class="section level1">
<h1 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>SessionInfo</h1>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.3 (2020-10-10)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.7</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] PhosR_1.1.5      BiocStyle_2.18.1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] colorspace_2.0-0            ggsignif_0.6.0             </span>
<span class="co">#&gt;   [3] ellipsis_0.3.1              class_7.3-17               </span>
<span class="co">#&gt;   [5] rio_0.5.16                  rprojroot_2.0.2            </span>
<span class="co">#&gt;   [7] circlize_0.4.12             markdown_1.1               </span>
<span class="co">#&gt;   [9] XVector_0.30.0              GenomicRanges_1.42.0       </span>
<span class="co">#&gt;  [11] GlobalOptions_0.1.2         ggdendro_0.1.22            </span>
<span class="co">#&gt;  [13] fs_1.5.0                    gridtext_0.1.4             </span>
<span class="co">#&gt;  [15] ggtext_0.1.1                ggpubr_0.4.0               </span>
<span class="co">#&gt;  [17] farver_2.0.3                xml2_1.3.2                 </span>
<span class="co">#&gt;  [19] cachem_1.0.4                knitr_1.31                 </span>
<span class="co">#&gt;  [21] broom_0.7.4                 pheatmap_1.0.12            </span>
<span class="co">#&gt;  [23] BiocManager_1.30.10         compiler_4.0.3             </span>
<span class="co">#&gt;  [25] backports_1.2.1             assertthat_0.2.1           </span>
<span class="co">#&gt;  [27] Matrix_1.2-18               fastmap_1.1.0              </span>
<span class="co">#&gt;  [29] limma_3.46.0                htmltools_0.5.1.1          </span>
<span class="co">#&gt;  [31] tools_4.0.3                 igraph_1.2.6               </span>
<span class="co">#&gt;  [33] gtable_0.3.0                glue_1.4.2                 </span>
<span class="co">#&gt;  [35] GenomeInfoDbData_1.2.4      reshape2_1.4.4             </span>
<span class="co">#&gt;  [37] dplyr_1.0.4                 Rcpp_1.0.6                 </span>
<span class="co">#&gt;  [39] carData_3.0-4               Biobase_2.50.0             </span>
<span class="co">#&gt;  [41] cellranger_1.1.0            pkgdown_1.6.1              </span>
<span class="co">#&gt;  [43] vctrs_0.3.6                 preprocessCore_1.52.1      </span>
<span class="co">#&gt;  [45] xfun_0.21                   stringr_1.4.0              </span>
<span class="co">#&gt;  [47] network_1.16.1              openxlsx_4.2.3             </span>
<span class="co">#&gt;  [49] lifecycle_0.2.0             rstatix_0.7.0              </span>
<span class="co">#&gt;  [51] dendextend_1.14.0           zlibbioc_1.36.0            </span>
<span class="co">#&gt;  [53] MASS_7.3-53                 scales_1.1.1               </span>
<span class="co">#&gt;  [55] ragg_0.4.1                  pcaMethods_1.82.0          </span>
<span class="co">#&gt;  [57] hms_1.0.0                   MatrixGenerics_1.2.1       </span>
<span class="co">#&gt;  [59] parallel_4.0.3              SummarizedExperiment_1.20.0</span>
<span class="co">#&gt;  [61] RColorBrewer_1.1-2          yaml_2.2.1                 </span>
<span class="co">#&gt;  [63] curl_4.3                    memoise_2.0.0              </span>
<span class="co">#&gt;  [65] gridExtra_2.3               ggplot2_3.3.3              </span>
<span class="co">#&gt;  [67] reshape_0.8.8               stringi_1.5.3              </span>
<span class="co">#&gt;  [69] highr_0.8                   S4Vectors_0.28.1           </span>
<span class="co">#&gt;  [71] desc_1.2.0                  e1071_1.7-4                </span>
<span class="co">#&gt;  [73] BiocGenerics_0.36.0         zip_2.1.1                  </span>
<span class="co">#&gt;  [75] shape_1.4.5                 GenomeInfoDb_1.26.2        </span>
<span class="co">#&gt;  [77] rlang_0.4.10                pkgconfig_2.0.3            </span>
<span class="co">#&gt;  [79] systemfonts_1.0.0           matrixStats_0.58.0         </span>
<span class="co">#&gt;  [81] bitops_1.0-6                evaluate_0.14              </span>
<span class="co">#&gt;  [83] lattice_0.20-41             purrr_0.3.4                </span>
<span class="co">#&gt;  [85] ruv_0.9.7.1                 labeling_0.4.2             </span>
<span class="co">#&gt;  [87] cowplot_1.1.1               tidyselect_1.1.0           </span>
<span class="co">#&gt;  [89] GGally_2.1.0                plyr_1.8.6                 </span>
<span class="co">#&gt;  [91] magrittr_2.0.1              bookdown_0.21              </span>
<span class="co">#&gt;  [93] R6_2.5.0                    IRanges_2.24.1             </span>
<span class="co">#&gt;  [95] generics_0.1.0              DelayedArray_0.16.1        </span>
<span class="co">#&gt;  [97] pillar_1.4.7                haven_2.3.1                </span>
<span class="co">#&gt;  [99] foreign_0.8-80              abind_1.4-5                </span>
<span class="co">#&gt; [101] RCurl_1.98-1.2              tibble_3.0.6               </span>
<span class="co">#&gt; [103] crayon_1.4.1                car_3.0-10                 </span>
<span class="co">#&gt; [105] rmarkdown_2.6               viridis_0.5.1              </span>
<span class="co">#&gt; [107] grid_4.0.3                  readxl_1.3.1               </span>
<span class="co">#&gt; [109] data.table_1.13.6           forcats_0.5.1              </span>
<span class="co">#&gt; [111] digest_0.6.27               tidyr_1.1.2                </span>
<span class="co">#&gt; [113] textshaping_0.2.1           stats4_4.0.3               </span>
<span class="co">#&gt; [115] munsell_0.5.0               viridisLite_0.3.0</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Pengyi Yang, Taiyun Kim, Hani Jieun Kim.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
