<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Pathway analysis with PhosR • PhosR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Pathway analysis with PhosR">
<meta property="og:description" content="PhosR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">PhosR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    PhosR
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/PhosR.html">Introduction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Preprocessing
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/web/imputation.html">Imputation</a>
    </li>
    <li>
      <a href="../../articles/web/batch_correction.html">Batch correction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Downstream analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/web/pathway_analysis.html">Pathway analysis</a>
    </li>
    <li>
      <a href="../../articles/web/site_gene_analysis.html">Site- and gene-centric analysis</a>
    </li>
    <li>
      <a href="../../articles/web/signalomes.html">Signalomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PYangLab/PhosR">
    <span class="fas fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="pathway_analysis_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Pathway analysis with PhosR</h1>
            
            <h4 class="date">17 March 2021</h4>
      
      
      <div class="hidden name"><code>pathway_analysis.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Most phosphoproteomic studies have adopted a phosphosite-level analysis of the data. To enable phosphoproteomic data analysis at the gene level, <code>PhosR</code> implements both site- and gene-centric analyses for detecting changes in kinase activities and signalling pathways through traditional enrichment analyses (over-representation or rank-based gene set test, together referred to as‘1-dimensional enrichment analysis’) as well as 2- and 3-dimensional analyses.</p>
<p>This vignette will perform gene-centric pathway enrichment analyses on the normalised myotube phosphoproteomic dataset using both over-representation and rank–based gene set tests and also provide an example of how <code>directPA</code> can be used to test which kinases are activated upon different stimulations in myotubes using 2-dimensional analyses <a href="https://academic.oup.com/bioinformatics/article/30/6/808/286146">Yang et al. 2014</a>.</p>
</div>
<div id="loading-packages-and-data" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-packages-and-data" class="anchor"></a>Loading packages and data</h1>
<p>First, we will load the PhosR package with few other packages will use for the demonstration purpose.</p>
<p>We will use RUV normalised L6 phosphopreteome data for demonstration of gene-centric pathway analysis. It contains phosphoproteome from three different treatment conditions: (1) AMPK agonist AICAR, (2) insulin (Ins), and (3) in combination (AICAR+Ins).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">calibrate</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma">limma</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">directPA</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Rn.eg.db</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">reactome.db</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">annotate</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PhosR</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"PhosphoSitePlus"</span><span class="op">)</span></code></pre></div>
<p>We will use the <code>ppe_RUV</code> matrix from <a href="https://pyanglab.github.io/PhosR/articles/batch_correction.html">batch_correction</a>.</p>
</div>
<div id="dimensional-enrichment-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#dimensional-enrichment-analysis" class="anchor"></a>1-dimensional enrichment analysis</h1>
<p>To enable enrichment analyses on both gene and phosphosite levels, PhosR implements a simple method called <code>phosCollapse</code> which reduces phosphosite level of information to the proteins for performing downstream gene-centric analyses. We will utilise two functions, <code>pathwayOverrepresent</code> and <code>pathwayRankBasedEnrichment</code>, to demonstrate 1-dimensional (over-representation and rank-based gene set test) gene-centric pathway enrichment analysis respectively.</p>
<p>First, extract phosphosite information from the ppe object.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sites</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">GeneSymbol</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,<span class="st">";"</span>,
                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Residue</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,
                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Site</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,
                 <span class="st">";"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span></code></pre></div>
<p>Then fit a linear model for each phosphosite.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_exp\\d"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span>
<span class="va">X</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">f</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">normalised</span>, <span class="va">X</span><span class="op">)</span></code></pre></div>
<p>Extract top-differentially regulated phosphosites for each condition compared to basal.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">table.AICAR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>, number<span class="op">=</span><span class="cn">Inf</span>, coef <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">table.Ins</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>, number<span class="op">=</span><span class="cn">Inf</span>, coef <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">table.AICARIns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>, number<span class="op">=</span><span class="cn">Inf</span>, coef <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>

<span class="va">DE1.RUV</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">table.AICAR</span><span class="op">[</span>,<span class="st">"adj.P.Val"</span><span class="op">]</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">table.Ins</span><span class="op">[</span>,<span class="st">"adj.P.Val"</span><span class="op">]</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">table.AICARIns</span><span class="op">[</span>,<span class="st">"adj.P.Val"</span><span class="op">]</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span>

<span class="co"># extract top-ranked phosphosites for each group comparison</span>
<span class="va">contrast.matrix1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/makeContrasts.html">makeContrasts</a></span><span class="op">(</span><span class="va">fAICARIns</span><span class="op">-</span><span class="va">fIns</span>, levels<span class="op">=</span><span class="va">X</span><span class="op">)</span>  <span class="co"># defining group comparisons</span>
<span class="va">contrast.matrix2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/makeContrasts.html">makeContrasts</a></span><span class="op">(</span><span class="va">fAICARIns</span><span class="op">-</span><span class="va">fAICAR</span>, levels<span class="op">=</span><span class="va">X</span><span class="op">)</span>  <span class="co"># defining group comparisons</span>
<span class="va">fit1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/contrasts.fit.html">contrasts.fit</a></span><span class="op">(</span><span class="va">fit</span>, <span class="va">contrast.matrix1</span><span class="op">)</span>
<span class="va">fit2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/contrasts.fit.html">contrasts.fit</a></span><span class="op">(</span><span class="va">fit</span>, <span class="va">contrast.matrix2</span><span class="op">)</span>
<span class="va">table.AICARInsVSIns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">fit1</span><span class="op">)</span>, number<span class="op">=</span><span class="cn">Inf</span><span class="op">)</span>
<span class="va">table.AICARInsVSAICAR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">fit2</span><span class="op">)</span>, number<span class="op">=</span><span class="cn">Inf</span><span class="op">)</span>

<span class="va">DE2.RUV</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">table.AICARInsVSIns</span><span class="op">[</span>,<span class="st">"adj.P.Val"</span><span class="op">]</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">table.AICARInsVSAICAR</span><span class="op">[</span>,<span class="st">"adj.P.Val"</span><span class="op">]</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span>

<span class="va">o</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">table.AICARInsVSIns</span><span class="op">)</span>
<span class="va">Tc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/cbind.html">cbind</a></span><span class="op">(</span><span class="va">table.Ins</span><span class="op">[</span><span class="va">o</span>,<span class="st">"logFC"</span><span class="op">]</span>, <span class="va">table.AICAR</span><span class="op">[</span><span class="va">o</span>,<span class="st">"logFC"</span><span class="op">]</span>, <span class="va">table.AICARIns</span><span class="op">[</span><span class="va">o</span>,<span class="st">"logFC"</span><span class="op">]</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sites</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">o</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"(.*)(;[A-Z])([0-9]+)(;)"</span>, <span class="st">"\\1;\\3;"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Ins"</span>, <span class="st">"AICAR"</span>, <span class="st">"AICAR+Ins"</span><span class="op">)</span></code></pre></div>
<p>Summarize phosphosite-level information to proteins for the downstream gene-centric analysis.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Tc.gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/phosCollapse.html">phosCollapse</a></span><span class="op">(</span><span class="va">Tc</span>, id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">";.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">)</span><span class="op">)</span>, 
                        stat<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">max</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"max"</span><span class="op">)</span>
<span class="va">geneSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="va">Tc.gene</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>, 
                        decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">Tc.gene</span><span class="op">)</span> <span class="op">*</span> <span class="fl">0.1</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">geneSet</span><span class="op">)</span>
<span class="co">#&gt; [1] "PPP1R13L" "SYNPO2L"  "AHNAK"    "USP10"    "SENP7"    "ATG2A"</span></code></pre></div>
</div>
<div id="prepare-the-reactome-annotation" class="section level1">
<h1 class="hasAnchor">
<a href="#prepare-the-reactome-annotation" class="anchor"></a>Prepare the Reactome annotation</h1>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pathways</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span><span class="op">(</span><span class="va">reactomePATHID2EXTID</span><span class="op">)</span>

<span class="va">path_names</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span><span class="op">(</span><span class="va">reactomePATHID2NAME</span><span class="op">)</span>
<span class="va">name_id</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">path_names</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">path_names</span><span class="op">)</span><span class="op">[</span><span class="va">name_id</span><span class="op">]</span>

<span class="va">pathways</span> <span class="op">=</span> <span class="va">pathways</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Rattus norvegicus"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span>, ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>

<span class="va">pathways</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">pathways</span>, <span class="kw">function</span><span class="op">(</span><span class="va">path</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">gene_name</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html">unname</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/annotate/man/getSYMBOL.html">getSYMBOL</a></span><span class="op">(</span><span class="va">path</span>, data <span class="op">=</span> <span class="st">"org.Rn.eg"</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/chartr.html">toupper</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">gene_name</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p>Perform 1D gene-centric pathway analysis</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">path1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/pathwayOverrepresent.html">pathwayOverrepresent</a></span><span class="op">(</span><span class="va">geneSet</span>, annotation<span class="op">=</span><span class="va">pathways</span>, 
                                universe <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tc.gene</span><span class="op">)</span>, alter <span class="op">=</span> <span class="st">"greater"</span><span class="op">)</span>
<span class="va">path2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/pathwayRankBasedEnrichment.html">pathwayRankBasedEnrichment</a></span><span class="op">(</span><span class="va">Tc.gene</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>, 
                                    annotation<span class="op">=</span><span class="va">pathways</span>, 
                                    alter <span class="op">=</span> <span class="st">"greater"</span><span class="op">)</span></code></pre></div>
<p>Next, we will compare enrichment of pathways (in negative log10 p-values) between the two 1-dimensional pathway enrichment analysis. On the scatter plot, the x-axis and y-axis refer to the p-values derived from the rank-based gene set test and over-representation test, respectively. We find several expected pathways, while these highly enriched pathways are largely in agreement between the two types of enrichment analyses.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lp1</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">path2</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="va">lp2</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">path1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">lp1</span>, <span class="va">lp2</span>, ylab<span class="op">=</span><span class="st">"Overrepresentation (-log10 pvalue)"</span>, xlab<span class="op">=</span><span class="st">"Rank-based enrichment (-log10 pvalue)"</span>, main<span class="op">=</span><span class="st">"Comparison of 1D pathway analyses"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span>

<span class="co"># select highly enriched pathways</span>
<span class="va">sel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">lp1</span> <span class="op">&gt;</span> <span class="fl">1.5</span> <span class="op">&amp;</span> <span class="va">lp2</span> <span class="op">&gt;</span> <span class="fl">0.9</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/calibrate/man/textxy.html">textxy</a></span><span class="op">(</span><span class="va">lp1</span><span class="op">[</span><span class="va">sel</span><span class="op">]</span>, <span class="va">lp2</span><span class="op">[</span><span class="va">sel</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_"</span>, <span class="st">" "</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"REACTOME_"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="va">sel</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="pathway_analysis_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
</div>
<div id="and-3-dimensional-signalling-pathway-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#and-3-dimensional-signalling-pathway-analysis" class="anchor"></a>2- and 3-dimensional signalling pathway analysis</h1>
<p>One key aspect in studying signalling pathways is to identify key kinases that are involved in signalling cascades. To identify these kinases, we make use of kinase-substrate annotation databases such as <code>PhosphoSitePlus</code> and <code>Phospho.ELM</code>. These databases are included in the <code>PhosR</code> and <code>directPA</code> packages already. To access them, simply load the package and access the data by <em>data(“PhosphoSitePlus”)</em> and <em>data(“PhosphoELM”)</em>.</p>
<p>The 2- and 3-dimensional analyses enable the investigation of kinases regulated by different combinations of treatments. We will introduce more advanced methods implemented in the R package <code>directPA</code> for performing “2 and 3-dimentional” direction site-centric kinase activity analyses.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># 2D direction site-centric kinase activity analyses</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="va">dpa1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/directPA/man/directPA.html">directPA</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span>, direction<span class="op">=</span><span class="fl">0</span>, 
                 annotation<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">PhosphoSite.rat</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">";[STY]"</span>, <span class="st">";"</span>, <span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>, 
                 main<span class="op">=</span><span class="st">"Direction pathway analysis"</span><span class="op">)</span>
<span class="va">dpa2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/directPA/man/directPA.html">directPA</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span>, direction<span class="op">=</span><span class="va">pi</span><span class="op">*</span><span class="fl">7</span><span class="op">/</span><span class="fl">4</span>, 
                 annotation<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">PhosphoSite.rat</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">";[STY]"</span>, <span class="st">";"</span>, <span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>, 
                 main<span class="op">=</span><span class="st">"Direction pathway analysis"</span><span class="op">)</span></code></pre></div>
<p><img src="pathway_analysis_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># top activated kinases</span>
<span class="va">dpa1</span><span class="op">$</span><span class="va">pathways</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span>
<span class="co">#&gt;        pvalue       size</span>
<span class="co">#&gt; AKT1   6.207001e-09 9   </span>
<span class="co">#&gt; MAPK1  0.00057404   9   </span>
<span class="co">#&gt; PRKACA 0.0006825021 25  </span>
<span class="co">#&gt; PRKAA1 0.000965093  6   </span>
<span class="co">#&gt; MAPK3  0.006670176  10</span>
<span class="va">dpa2</span><span class="op">$</span><span class="va">pathways</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span>
<span class="co">#&gt;         pvalue     size</span>
<span class="co">#&gt; PRKAA1  0.00463462 6   </span>
<span class="co">#&gt; AKT1    0.02942273 9   </span>
<span class="co">#&gt; CSNK2A1 0.2193148  12  </span>
<span class="co">#&gt; CDK5    0.2607434  5   </span>
<span class="co">#&gt; MAPK1   0.2767886  9</span></code></pre></div>
<p>There is also a function called <code>perturbPlot2d</code> implemented in <code>kinasePA</code> for testing and visualising activity of all kinases on all possible directions. Below are the demonstration from using this function.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">z1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/directPA/man/perturbPlot2d.html">perturbPlot2d</a></span><span class="op">(</span>Tc<span class="op">=</span><span class="va">Tc</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">]</span>, 
                    annotation<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">PhosphoSite.rat</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">";[STY]"</span>, <span class="st">";"</span>, <span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>,
                    cex<span class="op">=</span><span class="fl">1</span>, xlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>, <span class="fl">4</span><span class="op">)</span>, ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>, <span class="fl">4</span><span class="op">)</span>, 
                    main<span class="op">=</span><span class="st">"Kinase perturbation analysis"</span><span class="op">)</span></code></pre></div>
<p><img src="pathway_analysis_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">z2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/directPA/man/perturbPlot2d.html">perturbPlot2d</a></span><span class="op">(</span>Tc<span class="op">=</span><span class="va">Tc</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span>, annotation<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">PhosphoSite.rat</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">";[STY]"</span>, <span class="st">";"</span>, <span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>,
                    cex<span class="op">=</span><span class="fl">1</span>, xlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>, <span class="fl">4</span><span class="op">)</span>, ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>, <span class="fl">4</span><span class="op">)</span>, 
                    main<span class="op">=</span><span class="st">"Kinase perturbation analysis"</span><span class="op">)</span></code></pre></div>
<p><img src="pathway_analysis_files/figure-html/unnamed-chunk-13-2.png" width="700"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">z3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/directPA/man/perturbPlot2d.html">perturbPlot2d</a></span><span class="op">(</span>Tc<span class="op">=</span><span class="va">Tc</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span>, annotation<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">PhosphoSite.rat</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">";[STY]"</span>, <span class="st">";"</span>, <span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>,
                    cex<span class="op">=</span><span class="fl">1</span>, xlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>, <span class="fl">4</span><span class="op">)</span>, ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>, <span class="fl">4</span><span class="op">)</span>, 
                    main<span class="op">=</span><span class="st">"Kinase perturbation analysis"</span><span class="op">)</span></code></pre></div>
<p><img src="pathway_analysis_files/figure-html/unnamed-chunk-13-3.png" width="700"></p>
</div>
<div id="sessioninfo" class="section level1">
<h1 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>SessionInfo</h1>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.4 (2021-02-15)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.7</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] PhosR_1.1.6          annotate_1.68.0      XML_3.99-0.6        </span>
<span class="co">#&gt;  [4] reactome.db_1.74.0   org.Rn.eg.db_3.12.0  AnnotationDbi_1.52.0</span>
<span class="co">#&gt;  [7] IRanges_2.24.1       S4Vectors_0.28.1     Biobase_2.50.0      </span>
<span class="co">#&gt; [10] BiocGenerics_0.36.0  directPA_1.5         limma_3.46.0        </span>
<span class="co">#&gt; [13] calibrate_1.7.7      MASS_7.3-53         </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] colorspace_2.0-0            ggsignif_0.6.1             </span>
<span class="co">#&gt;   [3] ellipsis_0.3.1              class_7.3-18               </span>
<span class="co">#&gt;   [5] rio_0.5.26                  rprojroot_2.0.2            </span>
<span class="co">#&gt;   [7] circlize_0.4.12             XVector_0.30.0             </span>
<span class="co">#&gt;   [9] GenomicRanges_1.42.0        GlobalOptions_0.1.2        </span>
<span class="co">#&gt;  [11] ggdendro_0.1.22             fs_1.5.0                   </span>
<span class="co">#&gt;  [13] proxy_0.4-25                ggpubr_0.4.0               </span>
<span class="co">#&gt;  [15] bit64_4.0.5                 fansi_0.4.2                </span>
<span class="co">#&gt;  [17] cachem_1.0.4                knitr_1.31                 </span>
<span class="co">#&gt;  [19] jsonlite_1.7.2              broom_0.7.5                </span>
<span class="co">#&gt;  [21] pheatmap_1.0.12             BiocManager_1.30.10        </span>
<span class="co">#&gt;  [23] compiler_4.0.4              httr_1.4.2                 </span>
<span class="co">#&gt;  [25] backports_1.2.1             Matrix_1.3-2               </span>
<span class="co">#&gt;  [27] fastmap_1.1.0               htmltools_0.5.1.1          </span>
<span class="co">#&gt;  [29] tools_4.0.4                 igraph_1.2.6               </span>
<span class="co">#&gt;  [31] gtable_0.3.0                glue_1.4.2                 </span>
<span class="co">#&gt;  [33] GenomeInfoDbData_1.2.4      reshape2_1.4.4             </span>
<span class="co">#&gt;  [35] dplyr_1.0.5                 Rcpp_1.0.6                 </span>
<span class="co">#&gt;  [37] carData_3.0-4               cellranger_1.1.0           </span>
<span class="co">#&gt;  [39] jquerylib_0.1.3             pkgdown_1.6.1              </span>
<span class="co">#&gt;  [41] vctrs_0.3.6                 preprocessCore_1.52.1      </span>
<span class="co">#&gt;  [43] xfun_0.22                   stringr_1.4.0              </span>
<span class="co">#&gt;  [45] network_1.16.1              openxlsx_4.2.3             </span>
<span class="co">#&gt;  [47] lifecycle_1.0.0             rstatix_0.7.0              </span>
<span class="co">#&gt;  [49] dendextend_1.14.0           zlibbioc_1.36.0            </span>
<span class="co">#&gt;  [51] scales_1.1.1                BiocStyle_2.18.1           </span>
<span class="co">#&gt;  [53] pcaMethods_1.82.0           hms_1.0.0                  </span>
<span class="co">#&gt;  [55] ragg_1.1.1                  MatrixGenerics_1.2.1       </span>
<span class="co">#&gt;  [57] SummarizedExperiment_1.20.0 RColorBrewer_1.1-2         </span>
<span class="co">#&gt;  [59] yaml_2.2.1                  curl_4.3                   </span>
<span class="co">#&gt;  [61] memoise_2.0.0               gridExtra_2.3              </span>
<span class="co">#&gt;  [63] ggplot2_3.3.3               sass_0.3.1                 </span>
<span class="co">#&gt;  [65] reshape_0.8.8               stringi_1.5.3              </span>
<span class="co">#&gt;  [67] RSQLite_2.2.4               highr_0.8                  </span>
<span class="co">#&gt;  [69] desc_1.3.0                  e1071_1.7-5                </span>
<span class="co">#&gt;  [71] zip_2.1.1                   shape_1.4.5                </span>
<span class="co">#&gt;  [73] GenomeInfoDb_1.26.4         rlang_0.4.10               </span>
<span class="co">#&gt;  [75] pkgconfig_2.0.3             systemfonts_1.0.1          </span>
<span class="co">#&gt;  [77] matrixStats_0.58.0          bitops_1.0-6               </span>
<span class="co">#&gt;  [79] evaluate_0.14               lattice_0.20-41            </span>
<span class="co">#&gt;  [81] ruv_0.9.7.1                 purrr_0.3.4                </span>
<span class="co">#&gt;  [83] bit_4.0.4                   tidyselect_1.1.0           </span>
<span class="co">#&gt;  [85] GGally_2.1.1                plyr_1.8.6                 </span>
<span class="co">#&gt;  [87] magrittr_2.0.1              R6_2.5.0                   </span>
<span class="co">#&gt;  [89] generics_0.1.0              DelayedArray_0.16.2        </span>
<span class="co">#&gt;  [91] DBI_1.1.1                   haven_2.3.1                </span>
<span class="co">#&gt;  [93] pillar_1.5.1                foreign_0.8-81             </span>
<span class="co">#&gt;  [95] abind_1.4-5                 RCurl_1.98-1.3             </span>
<span class="co">#&gt;  [97] tibble_3.1.0                crayon_1.4.1               </span>
<span class="co">#&gt;  [99] car_3.0-10                  utf8_1.2.1                 </span>
<span class="co">#&gt; [101] rmarkdown_2.7               viridis_0.5.1              </span>
<span class="co">#&gt; [103] readxl_1.3.1                grid_4.0.4                 </span>
<span class="co">#&gt; [105] data.table_1.14.0           blob_1.2.1                 </span>
<span class="co">#&gt; [107] forcats_0.5.1               digest_0.6.27              </span>
<span class="co">#&gt; [109] xtable_1.8-4                tidyr_1.1.3                </span>
<span class="co">#&gt; [111] textshaping_0.3.1           munsell_0.5.0              </span>
<span class="co">#&gt; [113] viridisLite_0.3.0           bslib_0.2.4</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Pengyi Yang, Taiyun Kim, Hani Jieun Kim.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
