<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Signalome construction with PhosR â€¢ PhosR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Signalome construction with PhosR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">PhosR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    PhosR

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/PhosR.html">Introduction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Preprocessing

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/web/imputation.html">Imputation</a>
    </li>
    <li>
      <a href="../../articles/web/batch_correction.html">Batch correction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Downstream analysis

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/web/pathway_analysis.html">Pathway analysis</a>
    </li>
    <li>
      <a href="../../articles/web/site_gene_analysis.html">Site- and gene-centric analysis</a>
    </li>
    <li>
      <a href="../../articles/web/signalomes.html">Signalomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../reference/index.html">Functions</a>
</li>
<li>
  <a href="../../articles/web/PhosR_STAR_protocols.html">STAR Protocols</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PYangLab/PhosR" class="external-link">
    <span class="fa fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><script src="signalomes_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Signalome construction with PhosR</h1>
            
            <h4 data-toc-skip class="date">29 October 2024</h4>
      

      <div class="hidden name"><code>signalomes.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>A key component of the <code>PhosR</code> package is to construct signalomes. The signalome construction is composed of two main steps: 1) kinase-substrate relationsip scoring and 2) signalome construction. This involves a sequential workflow where the outputs of the first step are used as inputs of the latter step.</p>
<p>In brief, our kinase-substrate relationship scoring method (<code>kinaseSubstrateScore</code> and <code>kinaseSubstratePred</code>) prioritises potential kinases that could be responsible for the phosphorylation change of phosphosite on the basis of kinase recognition motif and phosphoproteomic dynamics. Using the kinase-substrate relationships derived from the scoring methods, we reconstruct signalome networks present in the data (<code>Signalomes</code>) wherin we highlight kinase regulation of discrete modules.</p>
</div>
<div class="section level2">
<h2 id="loading-packages-and-data">Loading packages and data<a class="anchor" aria-label="anchor" href="#loading-packages-and-data"></a>
</h2>
<p>First, we will load the <code>PhosR</code> package along with few other packages that we will be using in this section of the vignette.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">PhosR</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggobi.github.io/ggally/" class="external-link">GGally</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">ggpubr</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">calibrate</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://statnet.org/" class="external-link">network</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p>We will also be needing data containing kinase-substrate annotations from <code>PhosphoSitePlus</code>, kinase recognition motifs from <code>kinase motifs</code>, and annotations of kinase families from <code>kinase family</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"KinaseMotifs"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"KinaseFamily"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="setting-up-the-data">Setting up the data<a class="anchor" aria-label="anchor" href="#setting-up-the-data"></a>
</h2>
<p>As before, we will set up the data by cleaning up the phoshophosite labels and performing RUV normalisation. We will generate the <code>ppe_RUV</code> matrix as in <a href="https://pyanglab.github.io/PhosR/articles/batch_correction.html" class="external-link">batch_correction</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"phospho_L6_ratio_pe"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"SPSs"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"PhosphoSitePlus"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">##### Run batch correction</span></span>
<span><span class="va">ppe</span> <span class="op">&lt;-</span> <span class="va">phospho.L6.ratio.pe</span></span>
<span><span class="va">sites</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">GeneSymbol</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,<span class="st">";"</span>,</span>
<span>                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Residue</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,</span>
<span>                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Site</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,</span>
<span>                 <span class="st">";"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">grps</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"_.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">design</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">grps</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">ctl</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">sites</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">SPSs</span><span class="op">)</span></span>
<span><span class="va">ppe</span> <span class="op">=</span> <span class="fu"><a href="../../reference/RUVphospho.html">RUVphospho</a></span><span class="op">(</span><span class="va">ppe</span>, M <span class="op">=</span> <span class="va">design</span>, k <span class="op">=</span> <span class="fl">3</span>, ctl <span class="op">=</span> <span class="va">ctl</span><span class="op">)</span></span>
<span></span>
<span><span class="va">phosphoL6</span> <span class="op">=</span> <span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">normalised</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="generation-of-kinase-substrate-relationship-scores">Generation of kinase-substrate relationship scores<a class="anchor" aria-label="anchor" href="#generation-of-kinase-substrate-relationship-scores"></a>
</h2>
<p>Next, we will filtered for dynamically regulated phosphosites and then standardise the filtered matrix.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filter for up-regulated phosphosites</span></span>
<span><span class="va">phosphoL6.mean</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/meanAbundance.html">meanAbundance</a></span><span class="op">(</span><span class="va">phosphoL6</span>, grps <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"_.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">phosphoL6</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">aov</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/matANOVA.html">matANOVA</a></span><span class="op">(</span>mat<span class="op">=</span><span class="va">phosphoL6</span>, grps<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"_.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">phosphoL6</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">idx</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">aov</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">phosphoL6.mean</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">phosphoL6.reg</span> <span class="op">&lt;-</span> <span class="va">phosphoL6</span><span class="op">[</span><span class="va">idx</span>, ,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span>
<span></span>
<span><span class="va">L6.phos.std</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/standardise.html">standardise</a></span><span class="op">(</span><span class="va">phosphoL6.reg</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">L6.phos.std</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">GeneSymbol</span>, <span class="st">";"</span>, <span class="va">ppe</span><span class="op">@</span><span class="va">Residue</span>, <span class="va">ppe</span><span class="op">@</span><span class="va">Site</span>, <span class="st">";"</span><span class="op">)</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span></span></code></pre></div>
<p>We next extract the kinase recognition motifs from each phosphosite.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">L6.phos.seq</span> <span class="op">&lt;-</span> <span class="va">ppe</span><span class="op">@</span><span class="va">Sequence</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span></span></code></pre></div>
<p>Now that we have all the inputs for <code>kinaseSubstrateScore</code> and <code>kinaseSubstratePred</code> ready, we can proceed to the generation of kinase-substrate relationship scores.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">L6.matrices</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/kinaseSubstrateScore.html">kinaseSubstrateScore</a></span><span class="op">(</span>substrate.list <span class="op">=</span> <span class="va">PhosphoSite.mouse</span>, </span>
<span>                                    mat <span class="op">=</span> <span class="va">L6.phos.std</span>, seqs <span class="op">=</span> <span class="va">L6.phos.seq</span>, </span>
<span>                                    numMotif <span class="op">=</span> <span class="fl">5</span>, numSub <span class="op">=</span> <span class="fl">1</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="signalomes_files/figure-html/unnamed-chunk-7-1.png" width="768"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">L6.predMat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/kinaseSubstratePred.html">kinaseSubstratePred</a></span><span class="op">(</span><span class="va">L6.matrices</span>, top<span class="op">=</span><span class="fl">30</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </span></code></pre></div>
</div>
<div class="section level2">
<h2 id="signalome-construction">Signalome construction<a class="anchor" aria-label="anchor" href="#signalome-construction"></a>
</h2>
<p>The signalome construction uses the outputs of <code>kinaseSubstrateScore</code> and <code>kinaseSubstratePred</code> functions for the generation of a visualisation of the kinase regulation of discrete regulatory protein modules present in our phosphoproteomic data.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kinaseOI</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PRKAA1"</span>, <span class="st">"AKT1"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Signalomes_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/Signalomes.html">Signalomes</a></span><span class="op">(</span>KSR<span class="op">=</span><span class="va">L6.matrices</span>, </span>
<span>                                predMatrix<span class="op">=</span><span class="va">L6.predMat</span>, </span>
<span>                                exprsMat<span class="op">=</span><span class="va">L6.phos.std</span>, </span>
<span>                                KOI<span class="op">=</span><span class="va">kinaseOI</span><span class="op">)</span></span></code></pre></div>
<p><img src="signalomes_files/figure-html/unnamed-chunk-8-1.png" width="960"></p>
</div>
<div class="section level2">
<h2 id="generate-signalome-map">Generate signalome map<a class="anchor" aria-label="anchor" href="#generate-signalome-map"></a>
</h2>
<p>We can also visualise the relative contribution of each kinase towards the regulation of protein modules by plotting a balloon plot. In the balloon plot, the size of the balloons denote the percentage magnitude of kinase regulation in each module.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### generate palette</span></span>
<span><span class="va">my_color_palette</span> <span class="op">&lt;-</span> <span class="fu">grDevices</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Accent"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">kinase_all_color</span> <span class="op">&lt;-</span> <span class="fu">my_color_palette</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">L6.matrices</span><span class="op">$</span><span class="va">combinedScoreMatrix</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">kinase_all_color</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">L6.matrices</span><span class="op">$</span><span class="va">combinedScoreMatrix</span><span class="op">)</span></span>
<span><span class="va">kinase_signalome_color</span> <span class="op">&lt;-</span> <span class="va">kinase_all_color</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">L6.predMat</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="../../reference/plotSignalomeMap.html">plotSignalomeMap</a></span><span class="op">(</span>signalomes <span class="op">=</span> <span class="va">Signalomes_results</span>, color <span class="op">=</span> <span class="va">kinase_signalome_color</span><span class="op">)</span></span></code></pre></div>
<p><img src="signalomes_files/figure-html/unnamed-chunk-9-1.png" width="1440"></p>
</div>
<div class="section level2">
<h2 id="generate-signalome-network">Generate signalome network<a class="anchor" aria-label="anchor" href="#generate-signalome-network"></a>
</h2>
<p>Finally, we can also plot the signalome network that illustrates the connectivity between kinase signalome networks.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/plotKinaseNetwork.html">plotKinaseNetwork</a></span><span class="op">(</span>KSR <span class="op">=</span> <span class="va">L6.matrices</span>, predMatrix <span class="op">=</span> <span class="va">L6.predMat</span>, threshold <span class="op">=</span> <span class="fl">0.9</span>, color <span class="op">=</span> <span class="va">kinase_all_color</span><span class="op">)</span></span></code></pre></div>
<p><img src="signalomes_files/figure-html/unnamed-chunk-10-1.png" width="576"></p>
</div>
<div class="section level2">
<h2 id="sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Sonoma 14.7</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] network_1.18.2  calibrate_1.7.7 MASS_7.3-60.2   ggpubr_0.6.0   </span></span>
<span><span class="co">#&gt; [5] GGally_2.2.1    ggplot2_3.5.1   dplyr_1.1.4     PhosR_1.13.1   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] bitops_1.0-9                gridExtra_2.3              </span></span>
<span><span class="co">#&gt;   [3] rlang_1.1.4                 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [5] matrixStats_1.4.1           e1071_1.7-16               </span></span>
<span><span class="co">#&gt;   [7] compiler_4.4.1              systemfonts_1.1.0          </span></span>
<span><span class="co">#&gt;   [9] vctrs_0.6.5                 reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;  [11] stringr_1.5.1               pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [13] shape_1.4.6.1               crayon_1.5.3               </span></span>
<span><span class="co">#&gt;  [15] fastmap_1.2.0               backports_1.5.0            </span></span>
<span><span class="co">#&gt;  [17] XVector_0.42.0              labeling_0.4.3             </span></span>
<span><span class="co">#&gt;  [19] utf8_1.2.4                  rmarkdown_2.28             </span></span>
<span><span class="co">#&gt;  [21] preprocessCore_1.64.0       ragg_1.3.3                 </span></span>
<span><span class="co">#&gt;  [23] purrr_1.0.2                 xfun_0.48                  </span></span>
<span><span class="co">#&gt;  [25] zlibbioc_1.48.2             cachem_1.1.0               </span></span>
<span><span class="co">#&gt;  [27] GenomeInfoDb_1.38.8         jsonlite_1.8.9             </span></span>
<span><span class="co">#&gt;  [29] highr_0.11                  DelayedArray_0.28.0        </span></span>
<span><span class="co">#&gt;  [31] broom_1.0.7                 R6_2.5.1                   </span></span>
<span><span class="co">#&gt;  [33] stringi_1.8.4               bslib_0.8.0                </span></span>
<span><span class="co">#&gt;  [35] RColorBrewer_1.1-3          limma_3.58.1               </span></span>
<span><span class="co">#&gt;  [37] car_3.1-3                   GenomicRanges_1.54.1       </span></span>
<span><span class="co">#&gt;  [39] jquerylib_0.1.4             Rcpp_1.0.13                </span></span>
<span><span class="co">#&gt;  [41] SummarizedExperiment_1.32.0 knitr_1.48                 </span></span>
<span><span class="co">#&gt;  [43] IRanges_2.36.0              Matrix_1.7-0               </span></span>
<span><span class="co">#&gt;  [45] igraph_2.1.1                tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt;  [47] abind_1.4-8                 yaml_2.3.10                </span></span>
<span><span class="co">#&gt;  [49] viridis_0.6.5               lattice_0.22-6             </span></span>
<span><span class="co">#&gt;  [51] tibble_3.2.1                plyr_1.8.9                 </span></span>
<span><span class="co">#&gt;  [53] withr_3.0.2                 Biobase_2.62.0             </span></span>
<span><span class="co">#&gt;  [55] coda_0.19-4.1               evaluate_1.0.1             </span></span>
<span><span class="co">#&gt;  [57] desc_1.4.3                  ggstats_0.7.0              </span></span>
<span><span class="co">#&gt;  [59] proxy_0.4-27                circlize_0.4.16            </span></span>
<span><span class="co">#&gt;  [61] pillar_1.9.0                BiocManager_1.30.25        </span></span>
<span><span class="co">#&gt;  [63] MatrixGenerics_1.14.0       carData_3.0-5              </span></span>
<span><span class="co">#&gt;  [65] stats4_4.4.1                generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [67] RCurl_1.98-1.16             S4Vectors_0.40.2           </span></span>
<span><span class="co">#&gt;  [69] munsell_0.5.1               scales_1.3.0               </span></span>
<span><span class="co">#&gt;  [71] BiocStyle_2.30.0            class_7.3-22               </span></span>
<span><span class="co">#&gt;  [73] glue_1.8.0                  pheatmap_1.0.12            </span></span>
<span><span class="co">#&gt;  [75] tools_4.4.1                 dendextend_1.18.1          </span></span>
<span><span class="co">#&gt;  [77] ggsignif_0.6.4              fs_1.6.4                   </span></span>
<span><span class="co">#&gt;  [79] grid_4.4.1                  tidyr_1.3.1                </span></span>
<span><span class="co">#&gt;  [81] colorspace_2.1-1            GenomeInfoDbData_1.2.11    </span></span>
<span><span class="co">#&gt;  [83] Formula_1.2-5               cli_3.6.3                  </span></span>
<span><span class="co">#&gt;  [85] ruv_0.9.7.1                 textshaping_0.4.0          </span></span>
<span><span class="co">#&gt;  [87] fansi_1.0.6                 S4Arrays_1.2.1             </span></span>
<span><span class="co">#&gt;  [89] viridisLite_0.4.2           ggdendro_0.2.0             </span></span>
<span><span class="co">#&gt;  [91] pcaMethods_1.94.0           gtable_0.3.6               </span></span>
<span><span class="co">#&gt;  [93] rstatix_0.7.2               sass_0.4.9                 </span></span>
<span><span class="co">#&gt;  [95] digest_0.6.37               BiocGenerics_0.48.1        </span></span>
<span><span class="co">#&gt;  [97] SparseArray_1.2.4           farver_2.1.2               </span></span>
<span><span class="co">#&gt;  [99] htmlwidgets_1.6.4           htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt; [101] pkgdown_2.1.1               lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt; [103] GlobalOptions_0.1.2         statnet.common_4.10.0      </span></span>
<span><span class="co">#&gt; [105] statmod_1.5.0</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Pengyi Yang, Taiyun Kim, Hani Jieun Kim, Di Xiao.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
