<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Site- and gene-centric analysis • PhosR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Site- and gene-centric analysis">
<meta property="og:description" content="PhosR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">PhosR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    PhosR
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/PhosR.html">Introduction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Preprocessing
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/web/imputation.html">Imputation</a>
    </li>
    <li>
      <a href="../../articles/web/batch_correction.html">Batch correction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Downstream analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/web/pathway_analysis.html">Pathway analysis</a>
    </li>
    <li>
      <a href="../../articles/web/site_gene_analysis.html">Site- and gene-centric analysis</a>
    </li>
    <li>
      <a href="../../articles/web/signalomes.html">Signalomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PYangLab/PhosR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="site_gene_analysis_files/accessible-code-block-0.0.1/empty-anchor.js"></script><link href="site_gene_analysis_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="site_gene_analysis_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Site- and gene-centric analysis</h1>
            
            <h4 class="date">8 December 2020</h4>
      
      
      <div class="hidden name"><code>site_gene_analysis.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>While 1, 2, and 3D pathway analyses are useful for data generated from experiments with different treatment/conditions, analysis designed for time-course data may be better suited to analysis experiments that profile multiple time points.</p>
<p>Here, we will apply <code>ClueR</code> which is an R package specifically designed for time-course proteomic and phosphoproteomic data analysis <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004403">Yang et al. 2015</a>.</p>
</div>
<div id="loading-packages-and-data" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-packages-and-data" class="anchor"></a>Loading packages and data</h1>
<p>We will load the PhosR package with few other packages we will use for this tutorial.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">parallel</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ClueR</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">reactome.db</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Mm.eg.db</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">annotate</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PhosR</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p>We will load a dataset integrated from two time-course datasets of early and intermediate insulin signalling in mouse liver upon insulin stimulation to demonstrate the time-course phosphoproteomic data analyses.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho_liverInsTC_RUV_pe"</span><span class="op">)</span>
<span class="va">ppe</span> <span class="op">&lt;-</span> <span class="va">phospho.liver.Ins.TC.ratio.RUV.pe</span>
<span class="va">ppe</span>
<span class="co">#&gt; class: PhosphoExperiment </span>
<span class="co">#&gt; dim: 800 90 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): Quantification</span>
<span class="co">#&gt; rownames(800): LARP7;256; SRSF10;131; ... SIK3;493; GSK3A;21;</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(90): Intensity.Liver_Ins_0s_Bio7 Intensity.Liver_Ins_0s_Bio8</span>
<span class="co">#&gt;   ... Intensity.Liver_Ins_10m_Bio5 Intensity.Liver_Ins_10m_Bio6</span>
<span class="co">#&gt; colData names(0):</span></code></pre></div>
</div>
<div id="gene-centric-analyses-of-the-liver-phosphoproteome-data" class="section level1">
<h1 class="hasAnchor">
<a href="#gene-centric-analyses-of-the-liver-phosphoproteome-data" class="anchor"></a>Gene-centric analyses of the liver phosphoproteome data</h1>
<p>Let us start with gene-centric analysis. Such analysis can be directly applied to proteomics data. It can also be applied to phosphoproteomic data by using the <code>phosCollapse</code> function to summarise phosphosite information to proteins.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># take grouping information</span>
<span class="va">grps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, <span class="st">"_"</span><span class="op">)</span>, 
            <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span>

<span class="co"># select differentially phosphorylated sites</span>
<span class="va">sites.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/matANOVA.html">matANOVA</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, 
                    <span class="va">grps</span><span class="op">)</span>
<span class="va">ppm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/meanAbundance.html">meanAbundance</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, <span class="va">grps</span><span class="op">)</span>
<span class="va">sel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="op">(</span><span class="va">sites.p</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">ppm</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span>
<span class="va">ppm_filtered</span> <span class="op">&lt;-</span> <span class="va">ppm</span><span class="op">[</span><span class="va">sel</span>,<span class="op">]</span>

<span class="co"># summarise phosphosites information into gene level</span>
<span class="va">ppm_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/phosCollapse.html">phosCollapse</a></span><span class="op">(</span><span class="va">ppm_filtered</span>, 
            <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">";.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ppm_filtered</span><span class="op">)</span><span class="op">)</span>, 
                stat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">ppm_filtered</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">max</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"max"</span><span class="op">)</span>

<span class="co"># perform ClueR to identify optimal number of clusters</span>
<span class="va">pathways</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span><span class="op">(</span><span class="va">reactomePATHID2EXTID</span><span class="op">)</span>

<span class="va">pathways</span> <span class="op">=</span> <span class="va">pathways</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"R-MMU"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span>, ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>

<span class="va">pathways</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">pathways</span>, <span class="kw">function</span><span class="op">(</span><span class="va">path</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">gene_name</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html">unname</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/annotate/man/getSYMBOL.html">getSYMBOL</a></span><span class="op">(</span><span class="va">path</span>, data <span class="op">=</span> <span class="st">"org.Mm.eg"</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/chartr.html">toupper</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">gene_name</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">RNGkind</a></span><span class="op">(</span><span class="st">"L'Ecuyer-CMRG"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">c1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ClueR/man/runClue.html">runClue</a></span><span class="op">(</span><span class="va">ppm_gene</span>, annotation<span class="op">=</span><span class="va">pathways</span>, 
            kRange <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">10</span><span class="op">)</span>, rep <span class="op">=</span> <span class="fl">5</span>, effectiveSize <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">100</span><span class="op">)</span>, 
            pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>

<span class="co"># Visualise the evaluation results</span>
<span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Success<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">c1</span><span class="op">$</span><span class="va">evlMat</span><span class="op">)</span>, Freq<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">10</span><span class="op">)</span>, each<span class="op">=</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span>
<span class="va">myplot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Freq</span>, y<span class="op">=</span><span class="va">Success</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">Freq</span><span class="op">)</span>, fill<span class="op">=</span><span class="st">"gray"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_smooth.html">stat_smooth</a></span><span class="op">(</span>method<span class="op">=</span><span class="st">"loess"</span>, colour<span class="op">=</span><span class="st">"red"</span>, size<span class="op">=</span><span class="fl">3</span>, span <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"# of cluster"</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Enrichment score"</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">myplot</span></code></pre></div>
<p><img src="site_gene_analysis_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">best</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ClueR/man/clustOptimal.html">clustOptimal</a></span><span class="op">(</span><span class="va">c1</span>, rep<span class="op">=</span><span class="fl">5</span>, mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>, visualize <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="site_gene_analysis_files/figure-html/unnamed-chunk-4-2.png" width="700"></p>
</div>
<div id="site-centric-analyses-of-the-liver-phosphoproteome-data" class="section level1">
<h1 class="hasAnchor">
<a href="#site-centric-analyses-of-the-liver-phosphoproteome-data" class="anchor"></a>Site-centric analyses of the liver phosphoproteome data</h1>
<p>Phosphosite-centric analyses will perform using kinase-substrate annotation information from PhosphoSitePlus.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">RNGkind</a></span><span class="op">(</span><span class="st">"L'Ecuyer-CMRG"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="va">PhosphoSite.mouse2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">kinase</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"(.*)(;[A-Z])([0-9]+;)"</span>, <span class="st">"\\1;\\3"</span>, <span class="va">kinase</span><span class="op">)</span>
<span class="op">}</span>, <span class="va">PhosphoSite.mouse</span><span class="op">)</span>

<span class="co"># perform ClueR to identify optimal number of clusters</span>
<span class="va">c3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ClueR/man/runClue.html">runClue</a></span><span class="op">(</span><span class="va">ppm_filtered</span>, annotation<span class="op">=</span><span class="va">PhosphoSite.mouse2</span>, kRange <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fl">10</span>, rep <span class="op">=</span> <span class="fl">5</span>, effectiveSize <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">100</span><span class="op">)</span>, pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>

<span class="co"># Visualise the evaluation results</span>
<span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Success<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">c3</span><span class="op">$</span><span class="va">evlMat</span><span class="op">)</span>, Freq<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">2</span><span class="op">:</span><span class="fl">10</span>, each<span class="op">=</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span>
<span class="va">myplot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Freq</span>, y<span class="op">=</span><span class="va">Success</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">Freq</span><span class="op">)</span>, fill<span class="op">=</span><span class="st">"gray"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_smooth.html">stat_smooth</a></span><span class="op">(</span>method<span class="op">=</span><span class="st">"loess"</span>, colour<span class="op">=</span><span class="st">"red"</span>, size<span class="op">=</span><span class="fl">3</span>, span <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"# of cluster"</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Enrichment score"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">myplot</span></code></pre></div>
<p><img src="site_gene_analysis_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="va">best</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ClueR/man/clustOptimal.html">clustOptimal</a></span><span class="op">(</span><span class="va">c3</span>, rep<span class="op">=</span><span class="fl">10</span>, mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>, visualize <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="site_gene_analysis_files/figure-html/unnamed-chunk-5-2.png" width="700"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># Finding enriched pathways from each cluster</span>
<span class="va">best</span><span class="op">$</span><span class="va">enrichList</span>
<span class="co">#&gt; $`cluster 1`</span>
<span class="co">#&gt;      kinase   pvalue                 size</span>
<span class="co">#&gt; [1,] "PRKACA" "0.000184676866298047" "5" </span>
<span class="co">#&gt;      substrates                                                </span>
<span class="co">#&gt; [1,] "NR1H3;196;|MARCKS;163;|PRKACA;339;|ITPR1;1755;|SIK3;493;"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`cluster 3`</span>
<span class="co">#&gt;      kinase         pvalue                 size</span>
<span class="co">#&gt; [1,] "Humphrey.Akt" "0.000162969329853963" "5" </span>
<span class="co">#&gt; [2,] "Yang.Akt"     "0.000165386907010959" "6" </span>
<span class="co">#&gt;      substrates                                                         </span>
<span class="co">#&gt; [1,] "TSC2;939;|PFKFB2;486;|FOXO3;252;|FOXO1;316;|GSK3A;21;"            </span>
<span class="co">#&gt; [2,] "AKT1S1;247;|TSC2;939;|PFKFB2;486;|FOXO3;252;|FOXO1;316;|GSK3A;21;"</span></code></pre></div>
</div>
<div id="sessioninfo" class="section level1">
<h1 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>SessionInfo</h1>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.3 (2020-10-10)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.7</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Random number generation:</span>
<span class="co">#&gt;  RNG:     L'Ecuyer-CMRG </span>
<span class="co">#&gt;  Normal:  Inversion </span>
<span class="co">#&gt;  Sample:  Rejection </span>
<span class="co">#&gt;  </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    parallel  stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] PhosR_1.1.2          annotate_1.68.0      XML_3.99-0.5        </span>
<span class="co">#&gt;  [4] org.Mm.eg.db_3.12.0  reactome.db_1.74.0   AnnotationDbi_1.52.0</span>
<span class="co">#&gt;  [7] IRanges_2.24.0       S4Vectors_0.28.0     Biobase_2.50.0      </span>
<span class="co">#&gt; [10] BiocGenerics_0.36.0  ClueR_1.4            e1071_1.7-4         </span>
<span class="co">#&gt; [13] ggplot2_3.3.2       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] colorspace_2.0-0            ggsignif_0.6.0             </span>
<span class="co">#&gt;   [3] ellipsis_0.3.1              class_7.3-17               </span>
<span class="co">#&gt;   [5] rio_0.5.16                  rprojroot_2.0.2            </span>
<span class="co">#&gt;   [7] circlize_0.4.11             XVector_0.30.0             </span>
<span class="co">#&gt;   [9] GenomicRanges_1.42.0        GlobalOptions_0.1.2        </span>
<span class="co">#&gt;  [11] ggdendro_0.1.22             fs_1.5.0                   </span>
<span class="co">#&gt;  [13] farver_2.0.3                ggpubr_0.4.0               </span>
<span class="co">#&gt;  [15] bit64_4.0.5                 splines_4.0.3              </span>
<span class="co">#&gt;  [17] knitr_1.30                  broom_0.7.2                </span>
<span class="co">#&gt;  [19] pheatmap_1.0.12             BiocManager_1.30.10        </span>
<span class="co">#&gt;  [21] compiler_4.0.3              httr_1.4.2                 </span>
<span class="co">#&gt;  [23] backports_1.2.0             assertthat_0.2.1           </span>
<span class="co">#&gt;  [25] Matrix_1.2-18               limma_3.46.0               </span>
<span class="co">#&gt;  [27] htmltools_0.5.0             tools_4.0.3                </span>
<span class="co">#&gt;  [29] igraph_1.2.6                gtable_0.3.0               </span>
<span class="co">#&gt;  [31] glue_1.4.2                  GenomeInfoDbData_1.2.4     </span>
<span class="co">#&gt;  [33] reshape2_1.4.4              dplyr_1.0.2                </span>
<span class="co">#&gt;  [35] Rcpp_1.0.5                  carData_3.0-4              </span>
<span class="co">#&gt;  [37] cellranger_1.1.0            pkgdown_1.6.1              </span>
<span class="co">#&gt;  [39] vctrs_0.3.5                 nlme_3.1-149               </span>
<span class="co">#&gt;  [41] preprocessCore_1.52.0       xfun_0.19                  </span>
<span class="co">#&gt;  [43] stringr_1.4.0               network_1.16.1             </span>
<span class="co">#&gt;  [45] openxlsx_4.2.3              lifecycle_0.2.0            </span>
<span class="co">#&gt;  [47] rstatix_0.6.0               dendextend_1.14.0          </span>
<span class="co">#&gt;  [49] zlibbioc_1.36.0             MASS_7.3-53                </span>
<span class="co">#&gt;  [51] scales_1.1.1                BiocStyle_2.18.1           </span>
<span class="co">#&gt;  [53] ragg_0.4.0                  pcaMethods_1.82.0          </span>
<span class="co">#&gt;  [55] hms_0.5.3                   MatrixGenerics_1.2.0       </span>
<span class="co">#&gt;  [57] SummarizedExperiment_1.20.0 RColorBrewer_1.1-2         </span>
<span class="co">#&gt;  [59] yaml_2.2.1                  curl_4.3                   </span>
<span class="co">#&gt;  [61] memoise_1.1.0               gridExtra_2.3              </span>
<span class="co">#&gt;  [63] reshape_0.8.8               stringi_1.5.3              </span>
<span class="co">#&gt;  [65] RSQLite_2.2.1               desc_1.2.0                 </span>
<span class="co">#&gt;  [67] zip_2.1.1                   shape_1.4.5                </span>
<span class="co">#&gt;  [69] GenomeInfoDb_1.26.1         rlang_0.4.9                </span>
<span class="co">#&gt;  [71] pkgconfig_2.0.3             systemfonts_0.3.2          </span>
<span class="co">#&gt;  [73] matrixStats_0.57.0          bitops_1.0-6               </span>
<span class="co">#&gt;  [75] evaluate_0.14               lattice_0.20-41            </span>
<span class="co">#&gt;  [77] ruv_0.9.7.1                 purrr_0.3.4                </span>
<span class="co">#&gt;  [79] labeling_0.4.2              bit_4.0.4                  </span>
<span class="co">#&gt;  [81] tidyselect_1.1.0            GGally_2.0.0               </span>
<span class="co">#&gt;  [83] plyr_1.8.6                  magrittr_2.0.1             </span>
<span class="co">#&gt;  [85] R6_2.5.0                    generics_0.1.0             </span>
<span class="co">#&gt;  [87] DelayedArray_0.16.0         DBI_1.1.0                  </span>
<span class="co">#&gt;  [89] mgcv_1.8-33                 pillar_1.4.7               </span>
<span class="co">#&gt;  [91] haven_2.3.1                 foreign_0.8-80             </span>
<span class="co">#&gt;  [93] withr_2.3.0                 abind_1.4-5                </span>
<span class="co">#&gt;  [95] RCurl_1.98-1.2              tibble_3.0.4               </span>
<span class="co">#&gt;  [97] crayon_1.3.4                car_3.0-10                 </span>
<span class="co">#&gt;  [99] rmarkdown_2.5               viridis_0.5.1              </span>
<span class="co">#&gt; [101] grid_4.0.3                  readxl_1.3.1               </span>
<span class="co">#&gt; [103] data.table_1.13.2           blob_1.2.1                 </span>
<span class="co">#&gt; [105] forcats_0.5.0               digest_0.6.27              </span>
<span class="co">#&gt; [107] xtable_1.8-4                tidyr_1.1.2                </span>
<span class="co">#&gt; [109] textshaping_0.2.1           munsell_0.5.0              </span>
<span class="co">#&gt; [111] viridisLite_0.3.0</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Pengyi Yang, Taiyun Kim, Hani Jieun Kim.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
