<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An introduction to PhosR package • PhosR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An introduction to PhosR package">
<meta property="og:description" content="PhosR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">PhosR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    PhosR
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/PhosR.html">Introduction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Preprocessing
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/web/imputation.html">Imputation</a>
    </li>
    <li>
      <a href="../articles/web/batch_correction.html">Batch correction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Downstream analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/web/pathway_analysis.html">Pathway analysis</a>
    </li>
    <li>
      <a href="../articles/web/site_gene_analysis.html">Site- and gene-centric analysis</a>
    </li>
    <li>
      <a href="../articles/web/signalomes.html">Signalomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../articles/web/PhosR_STAR_protocols.html">STAR Protocols</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PYangLab/PhosR">
    <span class="fas fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="PhosR_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>An introduction to PhosR package</h1>
                        <h4 class="author">Taiyun Kim</h4>
            <address class="author_afil">
      School of Mathematics and Statistics, The University of Sydney<br><h4 class="author">Hani Jieun Kim</h4>
            <address class="author_afil">
      School of Mathematics and Statistics, The University of Sydney<br><h4 class="author">Di Xiao</h4>
            <address class="author_afil">
      Computational Systems Biology Group, Children’s Medical Research Institute, Faculty of Medicine and Health, The University of Sydney<br><h4 class="author">Pengyi Yang</h4>
            <address class="author_afil">
      School of Mathematics and Statistics, The University of Sydney<br><div class="hidden name"><code>PhosR.Rmd</code></div>

    </address>
</address>
</address>
</address>
</div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><code>PhosR</code> is a package for the comprehensive analysis of phosphoproteomic data. There are two major components to PhosR: processing and downstream analysis. PhosR consists of various processing tools for phosphoproteomic data including filtering, imputation, normalisaton and batch correction, enabling integration of multiple phosphoproteomic datasets. Downstream analytical tools consists of site- and protein-centric pathway analysis to evaluate activities of kinases and signalling pathways, large-scale kinase-substrate annotation from dynamic phosphoproteomic profiling, and visualisation and construction of signalomes present in the phosphoproteomic data of interest.</p>
<p>Below is a schematic overview of main componenets of PhosR, categorised into two broad steps of data analytics - processing and downstream analysis.</p>
<div class="figure">
<img src="../inst/graphical_abstract.png" alt=""><p class="caption">Overview of PhosR methods</p>
</div>
<p>The purpose of this vignette is to illustrate some uses of <code>PhosR</code> and explain its key components.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>Install the latest development version from GitHub using the <code>devtools</code> package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">devtools</span><span class="op">)</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"PYangLab/PhosR"</span><span class="op">)</span></code></pre></div>
<p>To install the Bioconductor version of PhosR, enter the following to your R console.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#if (!requireNamespace("BiocManager", quietly = TRUE))</span>
<span class="co">#    install.packages("BiocManager")</span>
<span class="co">#</span>
<span class="co">#BiocManager::install("PhosR")</span></code></pre></div>
</div>
<div id="loading-packages-and-data" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-packages-and-data" class="anchor"></a>Loading packages and data</h1>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
    <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PhosR</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p>For demonstration purposes, we provide a rat L6 myotubes phosphoproteome dataset in our package. The data contains ratios of samples treated with AICAR, an analog of adenosine monophosphate that stimulates AMPK activity, insulin (Ins), or in combination (AICAR+Ins) with the basal condition. The raw data have been deposited to the <a href="http://proteomecentral.proteomexchange.org/cgi/GetDataset">ProteomeXchange Consortium</a> via the PRIDE partner repository under PRIDE: PXD019127.</p>
</div>
<div id="setting-up-the-phosphoexperiment-object" class="section level1">
<h1 class="hasAnchor">
<a href="#setting-up-the-phosphoexperiment-object" class="anchor"></a>Setting up the PhosphoExperiment object</h1>
<p>To increase compatibility of PhosR functions with diverse processed datasets, we have implemented a <code>PhosphoExperiment</code> (ppe) object based on the <code>SummarizedExperiment</code> class. To create the <code>PhosphoExperiment</code> object, you will need a quantification matrix where columns refer to cells and rows refer to sites.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho_L6_ratio"</span><span class="op">)</span>

<span class="va">ppe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PhosphoExperiment-class.html">PhosphoExperiment</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>Quantification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">phospho.L6.ratio</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in .se_to_pe(se, UniprotID, GeneSymbol, Site, Residue, Sequence, : GeneSymbol is not specified. This may affect subsequent analysis steps.</code></pre>
<pre><code>## Warning in .se_to_pe(se, UniprotID, GeneSymbol, Site, Residue, Sequence, : Site is not specified. This may affect subsequent analysis steps.</code></pre>
<pre><code>## Warning in .se_to_pe(se, UniprotID, GeneSymbol, Site, Residue, Sequence, : Sequence is not specified. This may affect subsequent analysis steps.</code></pre>
<pre><code>## Warning in .se_to_pe(se, UniprotID, GeneSymbol, Site, Residue, Sequence, : Residue is not specified. This may affect subsequent analysis steps.</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "PhosphoExperiment"
## attr(,"package")
## [1] "PhosR"</code></pre>
<p>Additional annotation labels for the sites should be provided alongside the matrix. For each phosphosite, these include gene symbol, residue, position of phosphosite residue in the amino acid chain, and flanking sequence information of the phosphosite. Additional information such as the localisation probability may also be included.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ppe</span><span class="op">@</span><span class="va">GeneSymbol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, <span class="st">";"</span><span class="op">)</span>, <span class="st">"[["</span>, <span class="fl">2</span><span class="op">)</span>
<span class="va">ppe</span><span class="op">@</span><span class="va">Residue</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"[0-9]"</span>,<span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, <span class="st">";"</span><span class="op">)</span>, <span class="st">"[["</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="va">ppe</span><span class="op">@</span><span class="va">Site</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"[A-Z]"</span>,<span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, <span class="st">";"</span><span class="op">)</span>, <span class="st">"[["</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">ppe</span><span class="op">@</span><span class="va">Sequence</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, <span class="st">";"</span><span class="op">)</span>, <span class="st">"[["</span>, <span class="fl">4</span><span class="op">)</span></code></pre></div>
<p>Collectively, we can set up our data as following.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ppe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PhosphoExperiment-class.html">PhosphoExperiment</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>Quantification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">phospho.L6.ratio</span><span class="op">)</span><span class="op">)</span>, 
                  Site <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"[A-Z]"</span>,<span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, <span class="st">";"</span><span class="op">)</span>, <span class="st">"[["</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
                  GeneSymbol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, <span class="st">";"</span><span class="op">)</span>, <span class="st">"[["</span>, <span class="fl">2</span><span class="op">)</span>,
                  Residue <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"[0-9]"</span>,<span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, <span class="st">";"</span><span class="op">)</span>, <span class="st">"[["</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>,
                  Sequence <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, <span class="st">";"</span><span class="op">)</span>, <span class="st">"[["</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ppe</span></code></pre></div>
<pre><code>## class: PhosphoExperiment 
## dim: 6660 12 
## metadata(0):
## assays(1): Quantification
## rownames(6660): Q6AYR1;TFG;S198;MSAFGLTDDQVSGPPSAPTEDRSGTPDSIAS
##   D3ZRN2;MED1;T1035;STGGSKSPGSSGRCQTPPGVATPPIPKITIQ ...
##   F1LYK3;LOC685707;S810;VKPPSIANLDKVNSNSLDLPSSSDTHASKVP
##   G3V7U4;LMNB1;S393;RKLLEGEEERLKLSPSPSSRVTVSRASSSRS
## rowData names(0):
## colnames(12): AICAR_exp1 AICAR_exp2 ... AICARIns_exp3 AICARIns_exp4
## colData names(0):</code></pre>
</div>
<div id="part-a--preprocessing" class="section level1">
<h1 class="hasAnchor">
<a href="#part-a--preprocessing" class="anchor"></a>Part A. Preprocessing</h1>
<div id="imputation" class="section level2">
<h2 class="hasAnchor">
<a href="#imputation" class="anchor"></a>Imputation</h2>
<div id="introduction-1" class="section level3">
<h3 class="hasAnchor">
<a href="#introduction-1" class="anchor"></a>Introduction</h3>
<p><code>PhosR</code> is a package for the all-rounded analysis of phosphoproteomic data from processing to downstream analysis. This vignette will provide a step-by-step workflow of how PhosR can be used to process and analyse a a panel of phosphoproteomic datasets. As one of the first steps of data processing in phosphoproteomic analysis, we will begin by performing filtering and imputation of phosphoproteomic data with <code>PhosR</code>.</p>
</div>
<div id="loading-packages-and-data-1" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-packages-and-data-1" class="anchor"></a>Loading packages and data</h3>
<p>First, we will load the PhosR package. If you already haven’t done so, please install PhosR as instructed in the main page.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PhosR</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<div id="setting-up-the-data" class="section level4">
<h4 class="hasAnchor">
<a href="#setting-up-the-data" class="anchor"></a>Setting up the data</h4>
<p>We assume that you will have the raw data processed using platforms frequently used for mass-spectrometry based proteomics such as MaxQuant. For demonstration purposes, we will take a parts of phosphoproteomic data generated by <a href="https://doi.org/10.1038/nbt.3327">Humphrey et al.</a> with accession number PXD001792. The dataset contains the phosphoproteomic quantifications of two mouse liver cell lines (Hepa1.6 and FL38B) that were treated with either PBS (mock) or insulin.</p>
<p>Let us load the PhosphoExperiment (ppe) object</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho.cells.Ins.pe"</span><span class="op">)</span>

<span class="va">ppe</span> <span class="op">&lt;-</span> <span class="va">phospho.cells.Ins.pe</span> 
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "PhosphoExperiment"
## attr(,"package")
## [1] "PhosR"</code></pre>
<p>A quick glance of the object.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ppe</span></code></pre></div>
<pre><code>## class: PhosphoExperiment 
## dim: 5000 24 
## metadata(0):
## assays(1): Quantification
## rownames(5000): Q7TPV4;MYBBP1A;S1321;PQSALPKKRARLSLVSRSPSLLQSGVKKRRV
##   Q3UR85;MYRF;S304;PARAPSPPWPPQGPLSPGTGSLPLSIARAQT ...
##   P28659-4;NA;S18;AFKLDFLPEMMVDHCSLNSSPVSKKMNGTLD
##   E9Q8I9;FRY;S1380;HNIELVDSRLLLPGSSPSSPEDEVKDREGEV
## rowData names(0):
## colnames(24): Intensity.FL83B_Control_1 Intensity.FL83B_Control_2 ...
##   Intensity.Hepa1.6_Ins_5 Intensity.Hepa1.6_Ins_6
## colData names(0):</code></pre>
<p>We will take the grouping information from <code>colnames</code> of our matrix.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">grps</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_[0-9]{1}"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>For each cell line, there are two conditions (Control vs Insulin-stimulated) and 6 replicates for each condition.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># FL38B</span>
<span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"Intensity."</span>, <span class="st">""</span>, <span class="va">grps</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span></code></pre></div>
<pre><code>##  [1] "FL83B_Control" "FL83B_Control" "FL83B_Control" "FL83B_Control"
##  [5] "FL83B_Control" "FL83B_Control" "FL83B_Ins"     "FL83B_Ins"    
##  [9] "FL83B_Ins"     "FL83B_Ins"     "FL83B_Ins"     "FL83B_Ins"</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Hepa1</span>
<span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"Intensity."</span>, <span class="st">""</span>, <span class="va">grps</span><span class="op">)</span><span class="op">[</span><span class="fl">13</span><span class="op">:</span><span class="fl">24</span><span class="op">]</span></code></pre></div>
<pre><code>##  [1] "Hepa1.6_Control" "Hepa1.6_Control" "Hepa1.6_Control" "Hepa1.6_Control"
##  [5] "Hepa1.6_Control" "Hepa1.6_Control" "Hepa1.6_Ins"     "Hepa1.6_Ins"    
##  [9] "Hepa1.6_Ins"     "Hepa1.6_Ins"     "Hepa1.6_Ins"     "Hepa1.6_Ins"</code></pre>
<p>Note that there are in total 24 samples and 5,000 phosphosites profiled.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 5000   24</code></pre>
</div>
</div>
<div id="filtering-of-phosphosites" class="section level3">
<h3 class="hasAnchor">
<a href="#filtering-of-phosphosites" class="anchor"></a>Filtering of phosphosites</h3>
<p>Next, we will perform some filtering of phosphosites so that only phosphosites with quantification for at least 50% of the replicates in at least one of the conditions are retained. For this filtering step, we use the <code>selectGrps</code> function. The filtering leaves us with 1,772 phosphosites.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ppe_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectGrps.html">selectGrps</a></span><span class="op">(</span><span class="va">ppe</span>, <span class="va">grps</span>, <span class="fl">0.5</span>, n<span class="op">=</span><span class="fl">1</span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">ppe_filtered</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 1772   24</code></pre>
<p><code>selectGrps</code> gives you the option to relax the threshold for filtering. The filtering threshold can therefore be optimized for each dataset.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># In cases where you have fewer replicates ( e.g.,triplicates), you may want to </span>
<span class="co"># select phosphosites quantified in 70% of replicates. </span>
<span class="va">ppe_filtered_v1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectGrps.html">selectGrps</a></span><span class="op">(</span><span class="va">ppe</span>, <span class="va">grps</span>, <span class="fl">0.7</span>, n<span class="op">=</span><span class="fl">1</span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">ppe_filtered_v1</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 1330   24</code></pre>
</div>
<div id="imputation-of-phosphosites" class="section level3">
<h3 class="hasAnchor">
<a href="#imputation-of-phosphosites" class="anchor"></a>Imputation of phosphosites</h3>
<p>We can proceed to imputation now that we have filtered for suboptimal phosphosites. To take advantage of data structure and experimental design, PhosR provides users with a lot of flexibility for imputation. There are three functions for imputation: <code>scImpute</code>,<code>tInmpute</code>, and <code>ptImpute</code>. Here, we will demonstrate the use of <code>scImpute</code> and <code>ptImpute</code>.</p>
<div id="site--and-condition-specific-imputation" class="section level4">
<h4 class="hasAnchor">
<a href="#site--and-condition-specific-imputation" class="anchor"></a>Site- and condition-specific imputation</h4>
<p>The <code>scImpute</code> function is used for site- and condition-specific imputation. A pre-defined thereshold is used to select phosphosites to impute. Phosphosites with missing values equal to or greater than a predefined value will be imputed by sampling from the empirical normal distribution constructed from the quantification values of phosphosites from the same condition.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">ppe_imputed_tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scImpute.html">scImpute</a></span><span class="op">(</span><span class="va">ppe_filtered</span>, <span class="fl">0.5</span>, <span class="va">grps</span><span class="op">)</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe_filtered</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<p>In the above example, only phosphosites that are quantified in more than 50% of samples from the same condition will be imputed.</p>
</div>
<div id="paired-tail-based-imputation" class="section level4">
<h4 class="hasAnchor">
<a href="#paired-tail-based-imputation" class="anchor"></a>Paired tail-based imputation</h4>
<p>We then perform paired tail-based imputation on the dataset imputed with <code>scImpute</code>. Paired tail-based imputation performs imputation of phosphosites that have missing values in <em>all</em> replicates in one condition (e.g. in <em>basal</em>) but not in another condition (e.g., in <em>stimulation</em>). This method of imputation ensures that we do not accidentally filter phosphosites that seemingly have low detection rate.</p>
<p>As for <code>scImpute</code>, we can set a predefined threshold to in another condition (e.g. <em>stimulation</em>), the tail-based imputation is applied to impute for the missing values in the first condition.</p>
<p>As for <code>scImpute</code>, we can set a predefined threshold to in another condition (e.g. <em>stimulation</em>), the tail-based imputation is applied to impute for the missing values in the first condition.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">ppe_imputed</span> <span class="op">&lt;-</span> <span class="va">ppe_imputed_tmp</span>
<span class="va">ppe_imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ptImpute.html">ptImpute</a></span><span class="op">(</span><span class="va">ppe_imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">7</span>,<span class="fl">12</span><span class="op">)</span><span class="op">]</span>, 
                                 <span class="va">ppe_imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span>, 
                                 percent1 <span class="op">=</span> <span class="fl">0.6</span>, percent2 <span class="op">=</span> <span class="fl">0</span>, paired <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code>## idx1: 12</code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ppe_imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">13</span>,<span class="fl">18</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ptImpute.html">ptImpute</a></span><span class="op">(</span><span class="va">ppe_imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">19</span>,<span class="fl">24</span><span class="op">)</span><span class="op">]</span>, 
                                     <span class="va">ppe_imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">13</span>,<span class="fl">18</span><span class="op">)</span><span class="op">]</span>, 
                                     percent1 <span class="op">=</span> <span class="fl">0.6</span>, percent2 <span class="op">=</span> <span class="fl">0</span>, 
                                     paired <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code>## idx1: 29</code></pre>
<p>Lastly, we perform normalisation of the filtered and imputed phosphoproteomic data.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ppe_imputed_scaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/medianScaling.html">medianScaling</a></span><span class="op">(</span><span class="va">ppe_imputed</span>, scale <span class="op">=</span> <span class="cn">FALSE</span>, assay <span class="op">=</span> <span class="st">"imputed"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="quantification-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#quantification-plots" class="anchor"></a>Quantification plots</h3>
<p>A useful function in <code>PhosR</code> is to visualize the percentage of quantified sites before and after filtering and imputation. The main inputs of <code>plotQC</code> are the quantification matrix, sample labels (equating the column names of the matrix), an integer indicating the panel to plot, and lastly, a color vector. To visualize the percentage of quantified sites, use the <code>plotQC</code> function and set <em>panel = “quantify”</em> to visualise bar plots of samples.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe_filtered</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe_filtered</span><span class="op">)</span>, 
        panel <span class="op">=</span> <span class="st">"quantify"</span>, grps <span class="op">=</span> <span class="va">grps</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe_imputed_scaled</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">scaled</span>, 
       labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe_imputed_scaled</span><span class="op">)</span>, panel <span class="op">=</span> <span class="st">"quantify"</span>, grps <span class="op">=</span> <span class="va">grps</span><span class="op">)</span>
<span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<p>By setting <em>panel = “dendrogram”</em>, we can visualise the results of unsupervised hierarchical clustering of samples as a dendrogram. The dendrogram demonstrates that imputation has improved the clustering of the samples so that replicates from the same conditions cluster together.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe_filtered</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, 
       labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe_filtered</span><span class="op">)</span>, panel <span class="op">=</span> <span class="st">"dendrogram"</span>, 
       grps <span class="op">=</span> <span class="va">grps</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe_imputed_scaled</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">scaled</span>, 
       labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe_imputed_scaled</span><span class="op">)</span>, 
       panel <span class="op">=</span> <span class="st">"dendrogram"</span>, grps <span class="op">=</span> <span class="va">grps</span><span class="op">)</span>
<span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
<p>We can now move onto the next step in the <code>PhosR</code> workflow: integration of datasets and batch correction.</p>
</div>
</div>
<div id="batch-correction" class="section level2">
<h2 class="hasAnchor">
<a href="#batch-correction" class="anchor"></a>Batch correction</h2>
<div id="introduction-2" class="section level3">
<h3 class="hasAnchor">
<a href="#introduction-2" class="anchor"></a>Introduction</h3>
<p>A common but largely unaddressed challenge in phosphoproteomic data analysis is to correct for batch effect. Without correcting for batch effect, it is often not possible to analyze datasets in an integrative manner. To perform data integration and batch effect correction, we identified a set of stably phosphorylated sites (SPSs) across a panel of phosphoproteomic datasets and, using these SPSs, implemented a wrapper function of RUV-III from the <code>ruv</code> package called <code>RUVphospho</code>.</p>
<p>Note that when the input data contains missing values, imputation should be performed before batch correction since RUV-III requires a complete data matrix. The imputed values are removed by default after normalisation but can be retained for downstream analysis if the users wish to use the imputed matrix. This vignette will provide an example of how PhosR can be used for batch correction.</p>
</div>
<div id="loading-packages-and-data-2" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-packages-and-data-2" class="anchor"></a>Loading packages and data</h3>
<p>If you haven’t already done so, load the PhosR package.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PhosR</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://stringr.tidyverse.org">stringr</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p>In this example, we will use L6 myotube phosphoproteome dataset (with accession number PXD019127) and the SPSs we identified from a panel of phosphoproteomic datasets (please refer to <a href="https://www.biorxiv.org/content/10.1101/2020.08.31.276329v2.full">our preprint</a> for the full list of the datasets used). The <code>SPSs</code> will be used as our <code>negative control</code> in RUV normalisation.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho_L6_ratio_pe"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"SPSs"</span><span class="op">)</span>

<span class="va">ppe</span> <span class="op">&lt;-</span> <span class="va">phospho.L6.ratio.pe</span>
<span class="va">ppe</span></code></pre></div>
<pre><code>## class: PhosphoExperiment 
## dim: 6654 12 
## metadata(0):
## assays(1): Quantification
## rownames(6654): D3ZNS8;AAAS;S495;THIPLYFVNAQFPRFSPVLGRAQEPPAGGGG
##   Q9R0Z7;AAGAB;S210;RSVGSAESCQCEQEPSPTAERTESLPGHRSG ...
##   D3ZG78;ZZEF1;S1516;SGPSAAEVSTAEEPSSPSTPTRRPPFTRGRL
##   D3ZG78;ZZEF1;S1535;PTRRPPFTRGRLRLLSFRSMEETRPVPTVKE
## rowData names(0):
## colnames(12): AICAR_exp1 AICAR_exp2 ... AICARIns_exp3 AICARIns_exp4
## colData names(0):</code></pre>
</div>
<div id="setting-up-the-data-1" class="section level3">
<h3 class="hasAnchor">
<a href="#setting-up-the-data-1" class="anchor"></a>Setting up the data</h3>
<p>The L6 myotube data contains phosphoproteomic samples from three treatment conditions each with quadruplicates. Myotube cells were treated with either AICAR or Insulin (Ins), which are both important modulators of the insulin signalling pathway, or both (AICARIns) before phosphoproteomic analysis.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"AICAR_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code>## [1] "AICAR_exp1" "AICAR_exp2" "AICAR_exp3" "AICAR_exp4"</code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"^Ins_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code>## [1] "Ins_exp1" "Ins_exp2" "Ins_exp3" "Ins_exp4"</code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"AICARIns_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code>## [1] "AICARIns_exp1" "AICARIns_exp2" "AICARIns_exp3" "AICARIns_exp4"</code></pre>
<p>Note that we have in total 6654 quantified phosphosites and 12 samples in total.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 6654   12</code></pre>
<p>We have already performed the relevant processing steps to generate a dense matrix. Please refer to the <code>imputation</code> page to perform filtering and imputation of phosphosites in order to generate a matrix without any missing values.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>We will extract phosphosite labels.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sites</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">GeneSymbol</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,<span class="st">";"</span>,
                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Residue</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,
                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Site</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,
                 <span class="st">";"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span></code></pre></div>
<p>Lastly, we will take the grouping information from <code>colnames</code> of our matrix.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># take the grouping information</span>
<span class="va">grps</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span>
<span class="va">grps</span></code></pre></div>
<pre><code>##  [1] "AICAR"    "AICAR"    "AICAR"    "AICAR"    "Ins"      "Ins"     
##  [7] "Ins"      "Ins"      "AICARIns" "AICARIns" "AICARIns" "AICARIns"</code></pre>
</div>
<div id="diagnosing-batch-effect" class="section level3">
<h3 class="hasAnchor">
<a href="#diagnosing-batch-effect" class="anchor"></a>Diagnosing batch effect</h3>
<p>There are a number of ways to diagnose batch effect. In <code>PhosR</code>, we make use of two visualisation methods to detect batch effect: dendrogram of hierarchical clustering and a principal component analysis (PCA) plot. We use the <code>plotQC</code> function we introduced in the <em>imputation</em> section of the vignette.</p>
<p>By setting <em>panel = “dendrogram”</em>, we can plot the dendrogram illustrating the results of unsupervised hierarchical clustering of our 12 samples. Clustering results of the samples demonstrate that there is a strong batch effect (where batch denoted as <em>expX</em>, where X refers to the batch number). This is particularly evident for samples from <em>Ins</em> and <em>AICARIns</em> treated conditions.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, panel <span class="op">=</span> <span class="st">"dendrogram"</span>, grps<span class="op">=</span><span class="va">grps</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Before batch correction"</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-28-1.png" width="700"></p>
<p>We can also visualise the samples in PCA space by setting <em>panel = “pca”</em>. The PCA plot demonstrates aggregation of samples by batch rather than treatment groups (each point represents a sample coloured by treatment condition). It has become clearer that even within the <em>AICAR</em> treated samples, there is some degree of batch effect as data points are separated between samples from batches 1 and 2 and those from batches 3 and 4.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, grps<span class="op">=</span><span class="va">grps</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, panel <span class="op">=</span> <span class="st">"pca"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Before batch correction"</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-29-1.png" width="700"></p>
</div>
<div id="correcting-batch-effect" class="section level3">
<h3 class="hasAnchor">
<a href="#correcting-batch-effect" class="anchor"></a>Correcting batch effect</h3>
<p>We have now diagnosed that our dataset exhibits batch effect that is driven by experiment runs for samples treated with three different conditions. To address this batch effect, we correct for this unwanted variation in the data by utilising our pre-defined SPSs as a negative control for <code>RUVphospho</code>.</p>
<p>First, we construct a design matrix by condition.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">design</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">grps</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">design</span></code></pre></div>
<pre><code>##    grpsAICAR grpsAICARIns grpsIns
## 1          1            0       0
## 2          1            0       0
## 3          1            0       0
## 4          1            0       0
## 5          0            0       1
## 6          0            0       1
## 7          0            0       1
## 8          0            0       1
## 9          0            1       0
## 10         0            1       0
## 11         0            1       0
## 12         0            1       0
## attr(,"assign")
## [1] 1 1 1
## attr(,"contrasts")
## attr(,"contrasts")$grps
## [1] "contr.treatment"</code></pre>
<p>We will then use the <code>RUVphospho</code> function to normalise the data. Besides the quantification matrix and the design matrix, there are two other important inputs to <code>RUVphospho</code>: 1) the <em>ctl</em> argument is an integer vector denoting the position of SPSs within the quantification matrix 2) <em>k</em> parameter is an integer denoting the expected number of experimental (e.g., <em>treatment</em>) groups within the data</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># phosphoproteomics data normalisation and batch correction using RUV</span>
<span class="va">ctl</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">sites</span> <span class="op">%in%</span> <span class="va">SPSs</span><span class="op">)</span>
<span class="va">ppe</span> <span class="op">=</span> <span class="fu"><a href="../reference/RUVphospho.html">RUVphospho</a></span><span class="op">(</span><span class="va">ppe</span>, M <span class="op">=</span> <span class="va">design</span>, k <span class="op">=</span> <span class="fl">3</span>, ctl <span class="op">=</span> <span class="va">ctl</span><span class="op">)</span></code></pre></div>
</div>
<div id="quality-control" class="section level3">
<h3 class="hasAnchor">
<a href="#quality-control" class="anchor"></a>Quality control</h3>
<p>As quality control, we will demonstrate and evaluate our normalisation method with hierarchical clustering and PCA plot using again <code>plotQC</code>. Both the hierarchical clustering and PCA results demonstrate the normalisation procedure in <code>PhosR</code> facilitates effective batch correction.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot after batch correction</span>
<span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, grps<span class="op">=</span><span class="va">grps</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, panel <span class="op">=</span> <span class="st">"dendrogram"</span> <span class="op">)</span>
<span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">normalised</span>, grps<span class="op">=</span><span class="va">grps</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, panel<span class="op">=</span><span class="st">"dendrogram"</span><span class="op">)</span>
<span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-32-1.png" width="700"></p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot after batch correction</span>
<span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, panel <span class="op">=</span> <span class="st">"pca"</span>, grps<span class="op">=</span><span class="va">grps</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Before Batch correction"</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">normalised</span>, grps<span class="op">=</span><span class="va">grps</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, panel<span class="op">=</span><span class="st">"pca"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"After Batch correction"</span><span class="op">)</span>
<span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, nrow <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-33-1.png" width="700"></p>
</div>
<div id="generating-spss" class="section level3">
<h3 class="hasAnchor">
<a href="#generating-spss" class="anchor"></a>Generating SPSs</h3>
<p>We note that the current <em>SPS</em> are derived from phosphoproteomic datasets derived from mouse cells. To enable users working on phosphoproteomic datasets derived from other species, we have developed a function <code>getSPS</code> that takes in multiple phosphoproteomics datasets stored as phosphoExperiment objects to generate set of <em>SPS</em>. Users can use their in-house or public datasets derived from the same speciies to generate species-specific <em>SPS</em>, which they can use to normalise their data.Below demonstrate the usage of the <code>getSPS</code> function.</p>
<p>First, load datasets to use for SPS generation.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho_L6_ratio_pe"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho.liver.Ins.TC.ratio.RUV.pe"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho.cells.Ins.pe"</span><span class="op">)</span>
<span class="va">ppe1</span> <span class="op">&lt;-</span> <span class="va">phospho.L6.ratio.pe</span>
<span class="va">ppe2</span> <span class="op">&lt;-</span> <span class="va">phospho.liver.Ins.TC.ratio.RUV.pe</span>
<span class="co"># ppe2 &lt;- phospho.L1.Ins.ratio.subset.pe</span>
<span class="va">ppe3</span> <span class="op">&lt;-</span> <span class="va">phospho.cells.Ins.pe</span></code></pre></div>
<p>Filter, impute, and transform ppe3 (other inputs have already been processed).</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">grp3</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">'_[0-9]{1}'</span>, <span class="st">''</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe3</span><span class="op">)</span><span class="op">)</span>
<span class="va">ppe3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectGrps.html">selectGrps</a></span><span class="op">(</span><span class="va">ppe3</span>, grps <span class="op">=</span> <span class="va">grp3</span>, <span class="fl">0.5</span>, n<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="va">ppe3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tImpute.html">tImpute</a></span><span class="op">(</span><span class="va">ppe3</span><span class="op">)</span>
<span class="va">FL83B.ratio</span> <span class="op">&lt;-</span> <span class="va">ppe3</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">imputed</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span><span class="op">(</span><span class="va">ppe3</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"FL83B_Control"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe3</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="va">Hepa.ratio</span> <span class="op">&lt;-</span> <span class="va">ppe3</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">imputed</span><span class="op">[</span>, <span class="fl">13</span><span class="op">:</span><span class="fl">24</span><span class="op">]</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span><span class="op">(</span><span class="va">ppe3</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">imputed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"Hepa1.6_Control"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe3</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="va">ppe3</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/cbind.html">cbind</a></span><span class="op">(</span><span class="va">FL83B.ratio</span>, <span class="va">Hepa.ratio</span><span class="op">)</span></code></pre></div>
<p>Generate inputs of <code>getSPS</code>.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ppe.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">ppe1</span>, <span class="va">ppe2</span>, <span class="va">ppe3</span><span class="op">)</span>
<span class="co"># ppe.list &lt;- list(ppe1, ppe3)</span>

<span class="va">cond.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>grp1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe1</span><span class="op">)</span><span class="op">)</span>,
                  grp2 <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_sub.html">str_sub</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe2</span><span class="op">)</span>, end<span class="op">=</span><span class="op">-</span><span class="fl">5</span><span class="op">)</span>,
                  grp3 <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_sub.html">str_sub</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe3</span><span class="op">)</span>, end<span class="op">=</span><span class="op">-</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Finally run <code>getSPS</code> to generate list of SPSs.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">inhouse_SPSs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getSPS.html">getSPS</a></span><span class="op">(</span><span class="va">ppe.list</span>, conds <span class="op">=</span> <span class="va">cond.list</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: there aren't enough overlappling sites</code></pre>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">inhouse_SPSs</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "TMPO;S67;"    "RBM5;S621;"   "SRRM1;S427;"  "THOC2;S1417;" "SSB;S92;"    
## [6] "PPIG;S252;"</code></pre>
<p>Run <code>RUVphospho</code> using the newly generateed SPSs. (<strong>NOTE</strong> that we do not expect the results to be reproducible as the exempler SPSs have been generated from dummy datasets that have been heavily filtered.)</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sites</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">GeneSymbol</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,<span class="st">";"</span>,
              <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Residue</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,
              <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Site</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>, <span class="st">";"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>

<span class="va">ctl</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">sites</span> <span class="op">%in%</span> <span class="va">inhouse_SPSs</span><span class="op">)</span>
<span class="va">ppe</span> <span class="op">=</span> <span class="fu"><a href="../reference/RUVphospho.html">RUVphospho</a></span><span class="op">(</span><span class="va">ppe</span>, M <span class="op">=</span> <span class="va">design</span>, k <span class="op">=</span> <span class="fl">3</span>, ctl <span class="op">=</span> <span class="va">ctl</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
<div id="part-b--downstream-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#part-b--downstream-analysis" class="anchor"></a>Part B. Downstream analysis</h1>
<div id="pathway-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#pathway-analysis" class="anchor"></a>Pathway analysis</h2>
<div id="introduction-3" class="section level3">
<h3 class="hasAnchor">
<a href="#introduction-3" class="anchor"></a>Introduction</h3>
<p>Most phosphoproteomic studies have adopted a phosphosite-level analysis of the data. To enable phosphoproteomic data analysis at the gene level, <code>PhosR</code> implements both site- and gene-centric analyses for detecting changes in kinase activities and signalling pathways through traditional enrichment analyses (over-representation or rank-based gene set test, together referred to as‘1-dimensional enrichment analysis’) as well as 2- and 3-dimensional analyses.</p>
<p>This vignette will perform gene-centric pathway enrichment analyses on the normalised myotube phosphoproteomic dataset using both over-representation and rank–based gene set tests and also provide an example of how <code>directPA</code> can be used to test which kinases are activated upon different stimulations in myotubes using 2-dimensional analyses <a href="https://academic.oup.com/bioinformatics/article/30/6/808/286146">Yang et al. 2014</a>.</p>
</div>
<div id="loading-packages-and-data-3" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-packages-and-data-3" class="anchor"></a>Loading packages and data</h3>
<p>First, we will load the PhosR package with few other packages will use for the demonstration purpose.</p>
<p>We will use RUV normalised L6 phosphopreteome data for demonstration of gene-centric pathway analysis. It contains phosphoproteome from three different treatment conditions: (1) AMPK agonist AICAR, (2) insulin (Ins), and (3) in combination (AICAR+Ins).</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">calibrate</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma">limma</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">directPA</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Rn.eg.db</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">reactome.db</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">annotate</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PhosR</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"PhosphoSitePlus"</span><span class="op">)</span></code></pre></div>
<p>We will use the <code>ppe_RUV</code> matrix from <a href="https://pyanglab.github.io/PhosR/articles/batch_correction.html">batch_correction</a>.</p>
</div>
<div id="dimensional-enrichment-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#dimensional-enrichment-analysis" class="anchor"></a>1-dimensional enrichment analysis</h3>
<p>To enable enrichment analyses on both gene and phosphosite levels, PhosR implements a simple method called <code>phosCollapse</code> which reduces phosphosite level of information to the proteins for performing downstream gene-centric analyses. We will utilise two functions, <code>pathwayOverrepresent</code> and <code>pathwayRankBasedEnrichment</code>, to demonstrate 1-dimensional (over-representation and rank-based gene set test) gene-centric pathway enrichment analysis respectively.</p>
<p>First, extract phosphosite information from the ppe object.</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sites</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">GeneSymbol</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,<span class="st">";"</span>,
                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Residue</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,
                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Site</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,
                 <span class="st">";"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span></code></pre></div>
<p>Then fit a linear model for each phosphosite.</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_exp\\d"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span>
<span class="va">X</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">f</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">normalised</span>, <span class="va">X</span><span class="op">)</span></code></pre></div>
<p>Extract top-differentially regulated phosphosites for each condition compared to basal.</p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">table.AICAR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>, number<span class="op">=</span><span class="cn">Inf</span>, coef <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">table.Ins</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>, number<span class="op">=</span><span class="cn">Inf</span>, coef <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">table.AICARIns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>, number<span class="op">=</span><span class="cn">Inf</span>, coef <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>


<span class="va">DE1.RUV</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">table.AICAR</span><span class="op">[</span>,<span class="st">"adj.P.Val"</span><span class="op">]</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">table.Ins</span><span class="op">[</span>,<span class="st">"adj.P.Val"</span><span class="op">]</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">table.AICARIns</span><span class="op">[</span>,<span class="st">"adj.P.Val"</span><span class="op">]</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span>

<span class="co"># extract top-ranked phosphosites for each group comparison</span>
<span class="va">contrast.matrix1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/makeContrasts.html">makeContrasts</a></span><span class="op">(</span><span class="va">fAICARIns</span><span class="op">-</span><span class="va">fIns</span>, levels<span class="op">=</span><span class="va">X</span><span class="op">)</span>  <span class="co"># defining group comparisons</span>
<span class="va">contrast.matrix2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/makeContrasts.html">makeContrasts</a></span><span class="op">(</span><span class="va">fAICARIns</span><span class="op">-</span><span class="va">fAICAR</span>, levels<span class="op">=</span><span class="va">X</span><span class="op">)</span>  <span class="co"># defining group comparisons</span>
<span class="va">fit1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/contrasts.fit.html">contrasts.fit</a></span><span class="op">(</span><span class="va">fit</span>, <span class="va">contrast.matrix1</span><span class="op">)</span>
<span class="va">fit2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/contrasts.fit.html">contrasts.fit</a></span><span class="op">(</span><span class="va">fit</span>, <span class="va">contrast.matrix2</span><span class="op">)</span>
<span class="va">table.AICARInsVSIns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">fit1</span><span class="op">)</span>, number<span class="op">=</span><span class="cn">Inf</span><span class="op">)</span>
<span class="va">table.AICARInsVSAICAR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">fit2</span><span class="op">)</span>, number<span class="op">=</span><span class="cn">Inf</span><span class="op">)</span>


<span class="va">DE2.RUV</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">table.AICARInsVSIns</span><span class="op">[</span>,<span class="st">"adj.P.Val"</span><span class="op">]</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">table.AICARInsVSAICAR</span><span class="op">[</span>,<span class="st">"adj.P.Val"</span><span class="op">]</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span>

<span class="va">o</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">table.AICARInsVSIns</span><span class="op">)</span>
<span class="va">Tc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/cbind.html">cbind</a></span><span class="op">(</span><span class="va">table.Ins</span><span class="op">[</span><span class="va">o</span>,<span class="st">"logFC"</span><span class="op">]</span>, <span class="va">table.AICAR</span><span class="op">[</span><span class="va">o</span>,<span class="st">"logFC"</span><span class="op">]</span>, <span class="va">table.AICARIns</span><span class="op">[</span><span class="va">o</span>,<span class="st">"logFC"</span><span class="op">]</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sites</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">o</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"(.*)(;[A-Z])([0-9]+)(;)"</span>, <span class="st">"\\1;\\3;"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Ins"</span>, <span class="st">"AICAR"</span>, <span class="st">"AICAR+Ins"</span><span class="op">)</span></code></pre></div>
<p>Summarize phosphosite-level information to proteins for the downstream gene-centric analysis.</p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Tc.gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/phosCollapse.html">phosCollapse</a></span><span class="op">(</span><span class="va">Tc</span>, id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">";.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">)</span><span class="op">)</span>, 
                        stat<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">max</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"max"</span><span class="op">)</span>
<span class="va">geneSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="va">Tc.gene</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>, 
                        decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">Tc.gene</span><span class="op">)</span> <span class="op">*</span> <span class="fl">0.1</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">geneSet</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "PPP1R13L" "SYNPO2L"  "AHNAK"    "USP10"    "SENP7"    "ATG2A"</code></pre>
</div>
<div id="prepare-the-reactome-annotation" class="section level3">
<h3 class="hasAnchor">
<a href="#prepare-the-reactome-annotation" class="anchor"></a>Prepare the Reactome annotation</h3>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pathways</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span><span class="op">(</span><span class="va">reactomePATHID2EXTID</span><span class="op">)</span>

<span class="va">path_names</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span><span class="op">(</span><span class="va">reactomePATHID2NAME</span><span class="op">)</span>
<span class="va">name_id</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">path_names</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">path_names</span><span class="op">)</span><span class="op">[</span><span class="va">name_id</span><span class="op">]</span>

<span class="va">pathways</span> <span class="op">=</span> <span class="va">pathways</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Rattus norvegicus"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span>, ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>

<span class="va">pathways</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">pathways</span>, <span class="kw">function</span><span class="op">(</span><span class="va">path</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">gene_name</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html">unname</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/annotate/man/getSYMBOL.html">getSYMBOL</a></span><span class="op">(</span><span class="va">path</span>, data <span class="op">=</span> <span class="st">"org.Rn.eg"</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/chartr.html">toupper</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">gene_name</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
</div>
<div id="perform-1d-gene-centric-pathway-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#perform-1d-gene-centric-pathway-analysis" class="anchor"></a>Perform 1D gene-centric pathway analysis</h3>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">path1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pathwayOverrepresent.html">pathwayOverrepresent</a></span><span class="op">(</span><span class="va">geneSet</span>, annotation<span class="op">=</span><span class="va">pathways</span>, 
                                universe <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tc.gene</span><span class="op">)</span>, alter <span class="op">=</span> <span class="st">"greater"</span><span class="op">)</span>
<span class="va">path2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pathwayRankBasedEnrichment.html">pathwayRankBasedEnrichment</a></span><span class="op">(</span><span class="va">Tc.gene</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>, 
                                    annotation<span class="op">=</span><span class="va">pathways</span>, 
                                    alter <span class="op">=</span> <span class="st">"greater"</span><span class="op">)</span></code></pre></div>
<p>Next, we will compare enrichment of pathways (in negative log10 p-values) between the two 1-dimensional pathway enrichment analysis. On the scatter plot, the x-axis and y-axis refer to the p-values derived from the rank-based gene set test and over-representation test, respectively. We find several expected pathways, while these highly enriched pathways are largely in agreement between the two types of enrichment analyses.</p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lp1</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">path2</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="va">lp2</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">path1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">lp1</span>, <span class="va">lp2</span>, ylab<span class="op">=</span><span class="st">"Overrepresentation (-log10 pvalue)"</span>, xlab<span class="op">=</span><span class="st">"Rank-based enrichment (-log10 pvalue)"</span>, main<span class="op">=</span><span class="st">"Comparison of 1D pathway analyses"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span>

<span class="co"># select highly enriched pathways</span>
<span class="va">sel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">lp1</span> <span class="op">&gt;</span> <span class="fl">1.5</span> <span class="op">&amp;</span> <span class="va">lp2</span> <span class="op">&gt;</span> <span class="fl">0.9</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/calibrate/man/textxy.html">textxy</a></span><span class="op">(</span><span class="va">lp1</span><span class="op">[</span><span class="va">sel</span><span class="op">]</span>, <span class="va">lp2</span><span class="op">[</span><span class="va">sel</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_"</span>, <span class="st">" "</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"REACTOME_"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="va">sel</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-48-1.png" width="700"></p>
</div>
<div id="and-3-dimensional-signalling-pathway-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#and-3-dimensional-signalling-pathway-analysis" class="anchor"></a>2- and 3-dimensional signalling pathway analysis</h3>
<p>One key aspect in studying signalling pathways is to identify key kinases that are involved in signalling cascades. To identify these kinases, we make use of kinase-substrate annotation databases such as <code>PhosphoSitePlus</code> and <code>Phospho.ELM</code>. These databases are included in the <code>PhosR</code> and <code>directPA</code> packages already. To access them, simply load the package and access the data by <em>data(“PhosphoSitePlus”)</em> and <em>data(“PhosphoELM”)</em>.</p>
<p>The 2- and 3-dimensional analyses enable the investigation of kinases regulated by different combinations of treatments. We will introduce more advanced methods implemented in the R package <code>directPA</code> for performing “2 and 3-dimentional” direction site-centric kinase activity analyses.</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># 2D direction site-centric kinase activity analyses</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="va">dpa1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/directPA/man/directPA.html">directPA</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span>, direction<span class="op">=</span><span class="fl">0</span>, 
                 annotation<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">PhosphoSite.rat</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">";[STY]"</span>, <span class="st">";"</span>, <span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>, 
                 main<span class="op">=</span><span class="st">"Direction pathway analysis"</span><span class="op">)</span>
<span class="va">dpa2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/directPA/man/directPA.html">directPA</a></span><span class="op">(</span><span class="va">Tc</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span>, direction<span class="op">=</span><span class="va">pi</span><span class="op">*</span><span class="fl">7</span><span class="op">/</span><span class="fl">4</span>, 
                 annotation<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">PhosphoSite.rat</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">";[STY]"</span>, <span class="st">";"</span>, <span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>, 
                 main<span class="op">=</span><span class="st">"Direction pathway analysis"</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-49-1.png" width="700"></p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># top activated kinases</span>
<span class="va">dpa1</span><span class="op">$</span><span class="va">pathways</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></code></pre></div>
<pre><code>##        pvalue       size
## AKT1   6.207001e-09 9   
## MAPK1  0.00057404   9   
## PRKACA 0.0006825021 25  
## PRKAA1 0.000965093  6   
## MAPK3  0.006670176  10</code></pre>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dpa2</span><span class="op">$</span><span class="va">pathways</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></code></pre></div>
<pre><code>##         pvalue     size
## PRKAA1  0.00463462 6   
## AKT1    0.02942273 9   
## CSNK2A1 0.2193148  12  
## CDK5    0.2607434  5   
## MAPK1   0.2767886  9</code></pre>
<p>There is also a function called <code>perturbPlot2d</code> implemented in <code>kinasePA</code> for testing and visualising activity of all kinases on all possible directions. Below are the demonstration from using this function.</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">z1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/directPA/man/perturbPlot2d.html">perturbPlot2d</a></span><span class="op">(</span>Tc<span class="op">=</span><span class="va">Tc</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">]</span>, 
                    annotation<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">PhosphoSite.rat</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">";[STY]"</span>, <span class="st">";"</span>, <span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>,
                    cex<span class="op">=</span><span class="fl">1</span>, xlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>, <span class="fl">4</span><span class="op">)</span>, ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>, <span class="fl">4</span><span class="op">)</span>, 
                    main<span class="op">=</span><span class="st">"Kinase perturbation analysis"</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-50-1.png" width="700"></p>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">z2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/directPA/man/perturbPlot2d.html">perturbPlot2d</a></span><span class="op">(</span>Tc<span class="op">=</span><span class="va">Tc</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span>, annotation<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">PhosphoSite.rat</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">";[STY]"</span>, <span class="st">";"</span>, <span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>,
                    cex<span class="op">=</span><span class="fl">1</span>, xlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>, <span class="fl">4</span><span class="op">)</span>, ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>, <span class="fl">4</span><span class="op">)</span>, 
                    main<span class="op">=</span><span class="st">"Kinase perturbation analysis"</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-50-2.png" width="700"></p>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">z3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/directPA/man/perturbPlot2d.html">perturbPlot2d</a></span><span class="op">(</span>Tc<span class="op">=</span><span class="va">Tc</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span>, annotation<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">PhosphoSite.rat</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">";[STY]"</span>, <span class="st">";"</span>, <span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>,
                    cex<span class="op">=</span><span class="fl">1</span>, xlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>, <span class="fl">4</span><span class="op">)</span>, ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>, <span class="fl">4</span><span class="op">)</span>, 
                    main<span class="op">=</span><span class="st">"Kinase perturbation analysis"</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-50-3.png" width="700"></p>
</div>
</div>
<div id="site--and-gene--centric-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#site--and-gene--centric-analysis" class="anchor"></a>Site- and gene- centric analysis</h2>
<div id="introduction-4" class="section level3">
<h3 class="hasAnchor">
<a href="#introduction-4" class="anchor"></a>Introduction</h3>
<p>While 1, 2, and 3D pathway analyses are useful for data generated from experiments with different treatment/conditions, analysis designed for time-course data may be better suited to analysis experiments that profile multiple time points.</p>
<p>Here, we will apply <code>ClueR</code> which is an R package specifically designed for time-course proteomic and phosphoproteomic data analysis <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004403">Yang et al. 2015</a>.</p>
</div>
<div id="loading-packages-and-data-4" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-packages-and-data-4" class="anchor"></a>Loading packages and data</h3>
<p>We will load the PhosR package with few other packages we will use for this tutorial.</p>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">parallel</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ClueR</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">reactome.db</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Mm.eg.db</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">annotate</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PhosR</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p>We will load a dataset integrated from two time-course datasets of early and intermediate insulin signalling in mouse liver upon insulin stimulation to demonstrate the time-course phosphoproteomic data analyses.</p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># data("phospho_liverInsTC_RUV_pe")</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho.liver.Ins.TC.ratio.RUV.pe"</span><span class="op">)</span>
<span class="va">ppe</span> <span class="op">&lt;-</span> <span class="va">phospho.liver.Ins.TC.ratio.RUV.pe</span>
<span class="va">ppe</span></code></pre></div>
<pre><code>## class: PhosphoExperiment 
## dim: 800 90 
## metadata(0):
## assays(1): Quantification
## rownames(800): LARP7;256; SRSF10;131; ... SIK3;493; GSK3A;21;
## rowData names(0):
## colnames(90): Intensity.Liver_Ins_0s_Bio7 Intensity.Liver_Ins_0s_Bio8
##   ... Intensity.Liver_Ins_10m_Bio5 Intensity.Liver_Ins_10m_Bio6
## colData names(0):</code></pre>
</div>
<div id="gene-centric-analyses-of-the-liver-phosphoproteome-data" class="section level3">
<h3 class="hasAnchor">
<a href="#gene-centric-analyses-of-the-liver-phosphoproteome-data" class="anchor"></a>Gene-centric analyses of the liver phosphoproteome data</h3>
<p>Let us start with gene-centric analysis. Such analysis can be directly applied to proteomics data. It can also be applied to phosphoproteomic data by using the <code>phosCollapse</code> function to summarise phosphosite information to proteins.</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># take grouping information</span>

<span class="va">grps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span>, <span class="st">"_"</span><span class="op">)</span>, 
            <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span>

<span class="co"># select differentially phosphorylated sites</span>
<span class="va">sites.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/matANOVA.html">matANOVA</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, 
                    <span class="va">grps</span><span class="op">)</span>
<span class="va">ppm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/meanAbundance.html">meanAbundance</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">Quantification</span>, <span class="va">grps</span><span class="op">)</span>
<span class="va">sel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="op">(</span><span class="va">sites.p</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">ppm</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span>
<span class="va">ppm_filtered</span> <span class="op">&lt;-</span> <span class="va">ppm</span><span class="op">[</span><span class="va">sel</span>,<span class="op">]</span>

<span class="co"># summarise phosphosites information into gene level</span>
<span class="va">ppm_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/phosCollapse.html">phosCollapse</a></span><span class="op">(</span><span class="va">ppm_filtered</span>, 
            <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">";.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ppm_filtered</span><span class="op">)</span><span class="op">)</span>, 
                stat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">ppm_filtered</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">max</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"max"</span><span class="op">)</span>

<span class="co"># perform ClueR to identify optimal number of clusters</span>
<span class="va">pathways</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span><span class="op">(</span><span class="va">reactomePATHID2EXTID</span><span class="op">)</span>

<span class="va">pathways</span> <span class="op">=</span> <span class="va">pathways</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"R-MMU"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span>, ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>

<span class="va">pathways</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">pathways</span>, <span class="kw">function</span><span class="op">(</span><span class="va">path</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">gene_name</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html">unname</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/annotate/man/getSYMBOL.html">getSYMBOL</a></span><span class="op">(</span><span class="va">path</span>, data <span class="op">=</span> <span class="st">"org.Mm.eg"</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/chartr.html">toupper</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">gene_name</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">RNGkind</a></span><span class="op">(</span><span class="st">"L'Ecuyer-CMRG"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">c1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ClueR/man/runClue.html">runClue</a></span><span class="op">(</span><span class="va">ppm_gene</span>, annotation<span class="op">=</span><span class="va">pathways</span>, 
            kRange <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">10</span><span class="op">)</span>, rep <span class="op">=</span> <span class="fl">5</span>, effectiveSize <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">100</span><span class="op">)</span>, 
            pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>

<span class="co"># Visualise the evaluation results</span>
<span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Success<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">c1</span><span class="op">$</span><span class="va">evlMat</span><span class="op">)</span>, Freq<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">10</span><span class="op">)</span>, each<span class="op">=</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span>
<span class="va">myplot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Freq</span>, y<span class="op">=</span><span class="va">Success</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">Freq</span><span class="op">)</span>, fill<span class="op">=</span><span class="st">"gray"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_smooth.html">stat_smooth</a></span><span class="op">(</span>method<span class="op">=</span><span class="st">"loess"</span>, colour<span class="op">=</span><span class="st">"red"</span>, size<span class="op">=</span><span class="fl">3</span>, span <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"# of cluster"</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Enrichment score"</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">myplot</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula 'y ~ x'</code></pre>
<p><img src="PhosR_files/figure-html/unnamed-chunk-53-1.png" width="700"></p>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">best</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ClueR/man/clustOptimal.html">clustOptimal</a></span><span class="op">(</span><span class="va">c1</span>, rep<span class="op">=</span><span class="fl">5</span>, mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>, visualize <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-53-2.png" width="700"></p>
</div>
<div id="site-centric-analyses-of-the-liver-phosphoproteome-data" class="section level3">
<h3 class="hasAnchor">
<a href="#site-centric-analyses-of-the-liver-phosphoproteome-data" class="anchor"></a>Site-centric analyses of the liver phosphoproteome data</h3>
<p>Phosphosite-centric analyses will perform using kinase-substrate annotation information from PhosphoSitePlus.</p>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">RNGkind</a></span><span class="op">(</span><span class="st">"L'Ecuyer-CMRG"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="va">PhosphoSite.mouse2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">kinase</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"(.*)(;[A-Z])([0-9]+;)"</span>, <span class="st">"\\1;\\3"</span>, <span class="va">kinase</span><span class="op">)</span>
<span class="op">}</span>, <span class="va">PhosphoSite.mouse</span><span class="op">)</span>

<span class="co"># perform ClueR to identify optimal number of clusters</span>
<span class="va">c3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ClueR/man/runClue.html">runClue</a></span><span class="op">(</span><span class="va">ppm_filtered</span>, annotation<span class="op">=</span><span class="va">PhosphoSite.mouse2</span>, kRange <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fl">10</span>, rep <span class="op">=</span> <span class="fl">5</span>, effectiveSize <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">100</span><span class="op">)</span>, pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>

<span class="co"># Visualise the evaluation results</span>
<span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Success<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">c3</span><span class="op">$</span><span class="va">evlMat</span><span class="op">)</span>, Freq<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">2</span><span class="op">:</span><span class="fl">10</span>, each<span class="op">=</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span>

<span class="va">myplot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Freq</span>, y<span class="op">=</span><span class="va">Success</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">Freq</span><span class="op">)</span>, fill<span class="op">=</span><span class="st">"gray"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_smooth.html">stat_smooth</a></span><span class="op">(</span>method<span class="op">=</span><span class="st">"loess"</span>, colour<span class="op">=</span><span class="st">"red"</span>, size<span class="op">=</span><span class="fl">3</span>, span <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"# of cluster"</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Enrichment score"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">myplot</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula 'y ~ x'</code></pre>
<p><img src="PhosR_files/figure-html/unnamed-chunk-54-1.png" width="700"></p>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="va">best</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ClueR/man/clustOptimal.html">clustOptimal</a></span><span class="op">(</span><span class="va">c3</span>, rep<span class="op">=</span><span class="fl">10</span>, mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>, visualize <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-54-2.png" width="700"></p>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Finding enriched pathways from each cluster</span>
<span class="va">best</span><span class="op">$</span><span class="va">enrichList</span></code></pre></div>
<pre><code>## $`cluster 1`
##      kinase   pvalue                 size
## [1,] "PRKACA" "0.000184676866298047" "5" 
##      substrates                                                
## [1,] "NR1H3;196;|MARCKS;163;|PRKACA;339;|ITPR1;1755;|SIK3;493;"
## 
## $`cluster 3`
##      kinase         pvalue                 size
## [1,] "Humphrey.Akt" "0.000162969329853963" "5" 
## [2,] "Yang.Akt"     "0.000165386907010959" "6" 
##      substrates                                                         
## [1,] "TSC2;939;|PFKFB2;486;|FOXO3;252;|FOXO1;316;|GSK3A;21;"            
## [2,] "AKT1S1;247;|TSC2;939;|PFKFB2;486;|FOXO3;252;|FOXO1;316;|GSK3A;21;"</code></pre>
</div>
</div>
<div id="signalomes" class="section level2">
<h2 class="hasAnchor">
<a href="#signalomes" class="anchor"></a>Signalomes</h2>
<div id="introduction-5" class="section level3">
<h3 class="hasAnchor">
<a href="#introduction-5" class="anchor"></a>Introduction</h3>
<p>Lastly, a key component of the <code>PhosR</code> package is to construct signalomes. The signalome construction is composed of two main steps: 1) kinase-substrate relationsip scoring and 2) signalome construction. This involves a sequential workflow where the outputs of the first step are used as inputs of the latter step.</p>
<p>In brief, our kinase-substrate relationship scoring method (<code>kinaseSubstrateScore</code> and <code>kinaseSubstratePred</code>) prioritises potential kinases that could be responsible for the phosphorylation change of phosphosite on the basis of kinase recognition motif and phosphoproteomic dynamics. Using the kinase-substrate relationships derived from the scoring methods, we reconstruct signalome networks present in the data (<code>Signalomes</code>) wherin we highlight kinase regulation of discrete modules.</p>
<p>##3 Loading packages and data</p>
<p>First, we will load the <code>PhosR</code> package along with few other packages that we will be using in this section of the vignette.</p>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>

  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PhosR</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggobi.github.io/ggally/">GGally</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/">ggpubr</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">calibrate</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://statnet.org/">network</a></span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p>We will also be needing data containing kinase-substrate annotations from <code>PhosphoSitePlus</code>, kinase recognition motifs from <code>kinase motifs</code>, and annotations of kinase families from <code>kinase family</code>.</p>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"KinaseMotifs"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"KinaseFamily"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho_L6_ratio_pe"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"SPSs"</span><span class="op">)</span>

<span class="va">ppe</span> <span class="op">=</span> <span class="va">phospho.L6.ratio.pe</span>
<span class="va">sites</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">GeneSymbol</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,<span class="st">";"</span>,
                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Residue</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,
                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Site</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,
                 <span class="st">";"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
<span class="va">grps</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span>
<span class="va">design</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">grps</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">ctl</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">sites</span> <span class="op">%in%</span> <span class="va">SPSs</span><span class="op">)</span>
<span class="va">ppe</span> <span class="op">=</span> <span class="fu"><a href="../reference/RUVphospho.html">RUVphospho</a></span><span class="op">(</span><span class="va">ppe</span>, M <span class="op">=</span> <span class="va">design</span>, k <span class="op">=</span> <span class="fl">3</span>, ctl <span class="op">=</span> <span class="va">ctl</span><span class="op">)</span></code></pre></div>
</div>
<div id="setting-up-the-data-2" class="section level3">
<h3 class="hasAnchor">
<a href="#setting-up-the-data-2" class="anchor"></a>Setting up the data</h3>
<p>As before, we will set up the data by cleaning up the phoshophosite labels and performing RUV normalisation. We will generate the <code>ppe_RUV</code> matrix as in <a href="https://pyanglab.github.io/PhosR/articles/batch_correction.html">batch_correction</a>.</p>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"phospho_L6_ratio"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"SPSs"</span><span class="op">)</span>

<span class="co">##### Run batch correction</span>
<span class="va">ppe</span> <span class="op">&lt;-</span> <span class="va">phospho.L6.ratio.pe</span>
<span class="va">sites</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">GeneSymbol</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,<span class="st">";"</span>,
                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Residue</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,
                 <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">Site</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="va">x</span><span class="op">)</span>,
                 <span class="st">";"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
<span class="va">grps</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">)</span><span class="op">)</span>
<span class="va">design</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">grps</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">ctl</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">sites</span> <span class="op">%in%</span> <span class="va">SPSs</span><span class="op">)</span>
<span class="va">ppe</span> <span class="op">=</span> <span class="fu"><a href="../reference/RUVphospho.html">RUVphospho</a></span><span class="op">(</span><span class="va">ppe</span>, M <span class="op">=</span> <span class="va">design</span>, k <span class="op">=</span> <span class="fl">3</span>, ctl <span class="op">=</span> <span class="va">ctl</span><span class="op">)</span>


<span class="va">phosphoL6</span> <span class="op">=</span> <span class="va">ppe</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">normalised</span></code></pre></div>
</div>
<div id="generation-of-kinase-substrate-relationship-scores" class="section level3">
<h3 class="hasAnchor">
<a href="#generation-of-kinase-substrate-relationship-scores" class="anchor"></a>Generation of kinase-substrate relationship scores</h3>
<p>Next, we will filtered for dynamically regulated phosphosites and then standardise the filtered matrix.</p>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># filter for up-regulated phosphosites</span>

<span class="va">phosphoL6.mean</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/meanAbundance.html">meanAbundance</a></span><span class="op">(</span><span class="va">phosphoL6</span>, grps <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">phosphoL6</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">aov</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/matANOVA.html">matANOVA</a></span><span class="op">(</span>mat<span class="op">=</span><span class="va">phosphoL6</span>, grps<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_.+"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">phosphoL6</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">idx</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">aov</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">phosphoL6.mean</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>
<span class="va">phosphoL6.reg</span> <span class="op">&lt;-</span> <span class="va">phosphoL6</span><span class="op">[</span><span class="va">idx</span>, ,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>

<span class="va">L6.phos.std</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/standardise.html">standardise</a></span><span class="op">(</span><span class="va">phosphoL6.reg</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">L6.phos.std</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">ppe</span><span class="op">@</span><span class="va">GeneSymbol</span>, <span class="st">";"</span>, <span class="va">ppe</span><span class="op">@</span><span class="va">Residue</span>, <span class="va">ppe</span><span class="op">@</span><span class="va">Site</span>, <span class="st">";"</span><span class="op">)</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span></code></pre></div>
<p>We next extract the kinase recognition motifs from each phosphosite.</p>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">L6.phos.seq</span> <span class="op">&lt;-</span> <span class="va">ppe</span><span class="op">@</span><span class="va">Sequence</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span></code></pre></div>
<p>Now that we have all the inputs for <code>kinaseSubstrateScore</code> and <code>kinaseSubstratePred</code> ready, we can proceed to the generation of kinase-substrate relationship scores.</p>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">L6.matrices</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/kinaseSubstrateScore.html">kinaseSubstrateScore</a></span><span class="op">(</span>substrate.list <span class="op">=</span> <span class="va">PhosphoSite.mouse</span>, 
                                    mat <span class="op">=</span> <span class="va">L6.phos.std</span>, seqs <span class="op">=</span> <span class="va">L6.phos.seq</span>, 
                                    numMotif <span class="op">=</span> <span class="fl">5</span>, numSub <span class="op">=</span> <span class="fl">1</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-60-1.png" width="480"></p>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="va">L6.predMat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/kinaseSubstratePred.html">kinaseSubstratePred</a></span><span class="op">(</span><span class="va">L6.matrices</span>, top<span class="op">=</span><span class="fl">30</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </code></pre></div>
</div>
<div id="signalome-construction" class="section level3">
<h3 class="hasAnchor">
<a href="#signalome-construction" class="anchor"></a>Signalome construction</h3>
<p>The signalome construction uses the outputs of <code>kinaseSubstrateScore</code> and <code>kinaseSubstratePred</code> functions for the generation of a visualisation of the kinase regulation of discrete regulatory protein modules present in our phosphoproteomic data.</p>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">kinaseOI</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"PRKAA1"</span>, <span class="st">"AKT1"</span><span class="op">)</span>

<span class="va">Signalomes_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Signalomes.html">Signalomes</a></span><span class="op">(</span>KSR<span class="op">=</span><span class="va">L6.matrices</span>, 
                                predMatrix<span class="op">=</span><span class="va">L6.predMat</span>, 
                                exprsMat<span class="op">=</span><span class="va">L6.phos.std</span>, 
                                KOI<span class="op">=</span><span class="va">kinaseOI</span><span class="op">)</span></code></pre></div>
<pre><code>## calculating optimal number of clusters...</code></pre>
<pre><code>## optimal number of clusters = 3</code></pre>
<pre><code>## Note: The second link end is drawn out of sector '1'.</code></pre>
<pre><code>## Note: The second link end is drawn out of sector '2'.</code></pre>
<pre><code>## Note: The first link end is drawn out of sector 'MAP2K1'.</code></pre>
<pre><code>## Note: The first link end is drawn out of sector 'MTOR'.</code></pre>
<pre><code>## Note: The first link end is drawn out of sector 'TBK1'.</code></pre>
<p><img src="PhosR_files/figure-html/unnamed-chunk-61-1.png" width="960"></p>
</div>
<div id="generate-signalome-map" class="section level3">
<h3 class="hasAnchor">
<a href="#generate-signalome-map" class="anchor"></a>Generate signalome map</h3>
<p>We can also visualise the relative contribution of each kinase towards the regulation of protein modules by plotting a balloon plot. In the balloon plot, the size of the balloons denote the percentage magnitude of kinase regulation in each module.</p>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### generate palette</span>
<span class="va">my_color_palette</span> <span class="op">&lt;-</span> <span class="fu">grDevices</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span><span class="op">(</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Accent"</span><span class="op">)</span><span class="op">)</span>
<span class="va">kinase_all_color</span> <span class="op">&lt;-</span> <span class="fu">my_color_palette</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">L6.matrices</span><span class="op">$</span><span class="va">combinedScoreMatrix</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">kinase_all_color</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">L6.matrices</span><span class="op">$</span><span class="va">combinedScoreMatrix</span><span class="op">)</span>
<span class="va">kinase_signalome_color</span> <span class="op">&lt;-</span> <span class="va">kinase_all_color</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">L6.predMat</span><span class="op">)</span><span class="op">]</span>

<span class="fu"><a href="../reference/plotSignalomeMap.html">plotSignalomeMap</a></span><span class="op">(</span>signalomes <span class="op">=</span> <span class="va">Signalomes_results</span>, color <span class="op">=</span> <span class="va">kinase_signalome_color</span><span class="op">)</span></code></pre></div>
<p><img src="PhosR_files/figure-html/unnamed-chunk-62-1.png" width="1440"></p>
</div>
<div id="generate-signalome-network" class="section level3">
<h3 class="hasAnchor">
<a href="#generate-signalome-network" class="anchor"></a>Generate signalome network</h3>
<p>Finally, we can also plot the signalome network that illustrates the connectivity between kinase signalome networks.</p>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotKinaseNetwork.html">plotKinaseNetwork</a></span><span class="op">(</span>KSR <span class="op">=</span> <span class="va">L6.matrices</span>, predMatrix <span class="op">=</span> <span class="va">L6.predMat</span>, threshold <span class="op">=</span> <span class="fl">0.9</span>, color <span class="op">=</span> <span class="va">kinase_all_color</span><span class="op">)</span></code></pre></div>
<pre><code>## Note: The second link end is drawn out of sector 'CAMK2A'.</code></pre>
<pre><code>## Note: The second link end is drawn out of sector 'IRAK1'.</code></pre>
<pre><code>## Note: The second link end is drawn out of sector 'MAP2K1'.</code></pre>
<pre><code>## Note: The second link end is drawn out of sector 'PRKCD'.</code></pre>
<pre><code>## Note: The second link end is drawn out of sector 'PRKCE'.</code></pre>
<pre><code>## Note: The second link end is drawn out of sector 'PRKCH'.</code></pre>
<p><img src="PhosR_files/figure-html/unnamed-chunk-63-1.png" width="480"></p>
</div>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb122"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.5 (2021-03-31)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.7
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## Random number generation:
##  RNG:     L'Ecuyer-CMRG 
##  Normal:  Inversion 
##  Sample:  Rejection 
##  
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] network_1.16.1       ggpubr_0.4.0         GGally_2.1.1        
##  [4] dplyr_1.0.5          org.Mm.eg.db_3.12.0  ClueR_1.4           
##  [7] e1071_1.7-6          annotate_1.68.0      XML_3.99-0.6        
## [10] reactome.db_1.74.0   org.Rn.eg.db_3.12.0  AnnotationDbi_1.52.0
## [13] IRanges_2.24.1       S4Vectors_0.28.1     Biobase_2.50.0      
## [16] BiocGenerics_0.36.0  directPA_1.5         limma_3.46.0        
## [19] calibrate_1.7.7      MASS_7.3-53.1        ggplot2_3.3.3       
## [22] stringr_1.4.0        PhosR_1.1.7          BiocStyle_2.18.1    
## 
## loaded via a namespace (and not attached):
##   [1] readxl_1.3.1                backports_1.2.1            
##   [3] circlize_0.4.12             systemfonts_1.0.1          
##   [5] plyr_1.8.6                  igraph_1.2.6               
##   [7] splines_4.0.5               GenomeInfoDb_1.26.7        
##   [9] digest_0.6.27               htmltools_0.5.1.1          
##  [11] viridis_0.6.0               fansi_0.4.2                
##  [13] magrittr_2.0.1              memoise_2.0.0              
##  [15] openxlsx_4.2.3              matrixStats_0.58.0         
##  [17] pkgdown_1.6.1               colorspace_2.0-0           
##  [19] blob_1.2.1                  textshaping_0.3.3          
##  [21] haven_2.4.0                 xfun_0.22                  
##  [23] crayon_1.4.1                RCurl_1.98-1.3             
##  [25] jsonlite_1.7.2              glue_1.4.2                 
##  [27] ruv_0.9.7.1                 gtable_0.3.0               
##  [29] zlibbioc_1.36.0             XVector_0.30.0             
##  [31] DelayedArray_0.16.3         car_3.0-10                 
##  [33] shape_1.4.5                 abind_1.4-5                
##  [35] scales_1.1.1                pheatmap_1.0.12            
##  [37] DBI_1.1.1                   rstatix_0.7.0              
##  [39] Rcpp_1.0.6                  viridisLite_0.4.0          
##  [41] xtable_1.8-4                gridtext_0.1.4             
##  [43] foreign_0.8-81              bit_4.0.4                  
##  [45] proxy_0.4-25                preprocessCore_1.52.1      
##  [47] httr_1.4.2                  RColorBrewer_1.1-2         
##  [49] ellipsis_0.3.1              pkgconfig_2.0.3            
##  [51] reshape_0.8.8               farver_2.1.0               
##  [53] sass_0.3.1                  utf8_1.2.1                 
##  [55] tidyselect_1.1.0            labeling_0.4.2             
##  [57] rlang_0.4.10                reshape2_1.4.4             
##  [59] munsell_0.5.0               cellranger_1.1.0           
##  [61] tools_4.0.5                 cachem_1.0.4               
##  [63] generics_0.1.0              RSQLite_2.2.6              
##  [65] broom_0.7.6                 evaluate_0.14              
##  [67] fastmap_1.1.0               ggdendro_0.1.22            
##  [69] yaml_2.2.1                  ragg_1.1.2                 
##  [71] knitr_1.32                  bit64_4.0.5                
##  [73] fs_1.5.0                    zip_2.1.1                  
##  [75] purrr_0.3.4                 dendextend_1.14.0          
##  [77] nlme_3.1-152                xml2_1.3.2                 
##  [79] compiler_4.0.5              curl_4.3                   
##  [81] ggsignif_0.6.1              tibble_3.1.0               
##  [83] bslib_0.2.4                 stringi_1.5.3              
##  [85] highr_0.8                   desc_1.3.0                 
##  [87] forcats_0.5.1               lattice_0.20-41            
##  [89] Matrix_1.3-2                markdown_1.1               
##  [91] vctrs_0.3.7                 pillar_1.6.0               
##  [93] lifecycle_1.0.0             BiocManager_1.30.12        
##  [95] jquerylib_0.1.3             GlobalOptions_0.1.2        
##  [97] data.table_1.14.0           cowplot_1.1.1              
##  [99] bitops_1.0-6                GenomicRanges_1.42.0       
## [101] R6_2.5.0                    pcaMethods_1.82.0          
## [103] bookdown_0.21               gridExtra_2.3              
## [105] rio_0.5.26                  SummarizedExperiment_1.20.0
## [107] rprojroot_2.0.2             withr_2.4.1                
## [109] GenomeInfoDbData_1.2.4      mgcv_1.8-34                
## [111] hms_1.0.0                   ggtext_0.1.1               
## [113] grid_4.0.5                  tidyr_1.1.3                
## [115] class_7.3-18                rmarkdown_2.7              
## [117] MatrixGenerics_1.2.1        carData_3.0-4</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Pengyi Yang, Taiyun Kim, Hani Jieun Kim.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
